<!DOCTYPE html>
<html>
<head>
<title>protocol.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>

<style>
/* Tomorrow Theme */
/* http://jmblog.github.com/color-themes-for-google-code-highlightjs */
/* Original theme - https://github.com/chriskempson/tomorrow-theme */

/* Tomorrow Comment */
.hljs-comment,
.hljs-quote {
	color: #8e908c;
}

/* Tomorrow Red */
.hljs-variable,
.hljs-template-variable,
.hljs-tag,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class,
.hljs-regexp,
.hljs-deletion {
	color: #c82829;
}

/* Tomorrow Orange */
.hljs-number,
.hljs-built_in,
.hljs-builtin-name,
.hljs-literal,
.hljs-type,
.hljs-params,
.hljs-meta,
.hljs-link {
	color: #f5871f;
}

/* Tomorrow Yellow */
.hljs-attribute {
	color: #eab700;
}

/* Tomorrow Green */
.hljs-string,
.hljs-symbol,
.hljs-bullet,
.hljs-addition {
	color: #718c00;
}

/* Tomorrow Blue */
.hljs-title,
.hljs-section {
	color: #4271ae;
}

/* Tomorrow Purple */
.hljs-keyword,
.hljs-selector-tag {
	color: #8959a8;
}

.hljs {
	display: block;
	overflow-x: auto;
	color: #4d4d4c;
	padding: 0.5em;
}

.hljs-emphasis {
	font-style: italic;
}

.hljs-strong {
	font-weight: bold;
}
</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>

<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="comparing-different-classification-tools-regarding-their-usability-for-metagenomic-long-read-data">Comparing different classification tools regarding their usability for metagenomic long-read data</h1>
<hr>
<h2 id="table-of-contents">Table of Contents</h2>
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#material-and-methods">Material and Methods</a>
<ul>
<li><a href="#data">Data</a></li>
<li><a href="#tools">Tools</a>
<ul>
<li><a href="#classification">Classification</a></li>
<li><a href="#others">Others</a></li>
</ul>
</li>
<li><a href="#metrics">Metrics</a>
<ul>
<li><a href="#area-under-precision-recall-curve">Area Under Precision-Recall Curve</a></li>
<li><a href="#abundance-profile-similarity">Abundance Profile Similarity</a></li>
<li><a href="#computational-requirements">Computational Requirements</a></li>
</ul>
</li>
<li><a href="#multi-locus-sequence-typing">Multi Locus Sequence Typing</a></li>
</ul>
</li>
<li><a href="#results-and-discussion">Results and Discussion</a>
<ul>
<li><a href="#comparison-using-the-metrics">Comparison Using the Metrics</a>
<ul>
<li><a href="#benchmarking-with-area-under-precision-recall-curve">Benchmarking with Area under Precision-Recall Curve</a></li>
<li><a href="#benchmarking-with-abundance-profile-similarity">Benchmarking with Abundance Profile Similarity</a>
<ul>
<li><a href="#differences-for-gram-positive-and-gram-negative-species">Differences for Gram-positive and Gram-negative species</a></li>
</ul>
</li>
<li><a href="#computational-requirements-runtime-and-database-creation">Computational Requirements: Runtime and Database Creation</a></li>
</ul>
</li>
<li><a href="#classification-validation-with-multi-locus-sequence-typing">Classification Validation with Multi Locus Sequence Typing</a></li>
<li><a href="#classification-results">Classification Results</a></li>
<li><a href="#difficulties-with-the-tools">Difficulties with the Tools</a></li>
</ul>
</li>
<li><a href="#conclusion">Conclusion</a></li>
<li><a href="#attachments-and-supplementary-information">Attachments and Supplementary Information</a>
<ul>
<li><a href="#tables">Tables</a></li>
<li><a href="#figures">Figures</a></li>
</ul>
</li>
<li><a href="#citations">Citations</a></li>
</ul>
<h1 id="introduction">Introduction</h1>
<p>Metagenomic analysis and sequencing gain more and more importance <a href="https://doi.org/10.1093/bib/bbs035" title="Li, W., Fu, L., Niu, B., Wu, S., &amp; Wooley, J. (2012). Ultrafast clustering algorithms for metagenomic sequence analysis. *Briefings in bioinformatics*, 13(6), 656-668.">[1]</a>. They enable the better understanding of functions, mechanisms and participants of environmental niches as well as the analysis of diversity and interactions of microbial species and host (e.g. human gut microbiome), but also the identification and analysis of microbial products <a href="https://doi.org/10.1007/s10311-020-01010-z" title="Datta, S., Rajnish, K. N., Samuel, M. S., Pugazlendhi, A., &amp; Selvarajan, E. (2020). Metagenomic applications in microbial diversity, bioremediation, pollution monitoring, enzyme and drug discovery. A review. *Environmental Chemistry Letters*, 18(4), 1229-1241.">[2]</a>.
Metagenomic analyses are of importance because, unlike other types of samples, they are not about characterising an isolated and extracted sample of a species, but, as mentioned before, about understanding a whole environment. Especially concerning different environments that can also affect humans, such as the human gut microbiome, the bacteria in milk cultures which are important for fermentation with their interaction, or different soil environments, the study of an entire population is advantageous. <br>
Furthermore, long reads are still a field of important research with Oxford Nanopore Technologies (ONT) publishing the first nanopore sequencer MinION <a href="https://doi.org/10.1016/j.gpb.2016.05.004" title="Lu, H., Giordano, F., &amp; Ning, Z. (2016). Oxford Nanopore MinION sequencing and genome assembly. *Genomics, proteomics &amp; bioinformatics*, 14(5), 265-279.">[3]</a>. Combining these two fields leads to the need for long-read metagenomic processing tools, first and foremost classification tools. Few approaches for short-read data are applicable without changes, since a common use of k-mers, for example, might introduce the problem of a fitting default length - is it possible and recommendable to use the same length as for short reads? Additionally, the longer sequences probably need more memory than shorter reads, leading to a new set of problems regarding the technical boundaries, some of them still being relevant for short reads. Hence, an overview of different classifiers and their usability of metagenomic long-read samples might be useful. <br></p>
<p>The difficulties of a benchmark between different classifiers lie in the different classification approaches, the different types as well as the different output formats. The different types regard the sequence type used for classification - there are for example DNA and protein sequences. This means, for protein types, the input (DNA) sequences are translated into all six open reading frame protein sequences if necessary. <br>
A common approach for classification is k-mer-based. Tools like Kraken2 <a href="https://doi.org/10.1186/s13059-019-1891-0" title="Wood, D. E., Lu, J., &amp; Langmead, B. (2019). Improved metagenomic analysis with Kraken 2. *Genome biology*, 20(1), 1-13.">[4]</a>, CLARK <a href="https://doi.org/10.1186/s12864-015-1419-2" title="Ounit, R., Wanamaker, S., Close, T. J., &amp; Lonardi, S. (2015). CLARK: fast and accurate classification of metagenomic and genomic sequences using discriminative k-mers. *BMC genomics*, 16(1), 1-13.">[5]</a>, and CCMetagen <a href="https://doi.org/10.1186/s13059-020-02014-2" title="Marcelino, V. R., Clausen, P. T., Buchmann, J. P., Wille, M., Iredell, J. R., Meyer, W., ... &amp; Holmes, E. C. (2020). CCMetagen: comprehensive and accurate identification of eukaryotes and prokaryotes in metagenomic data. *Genome biology*, 21, 1-15.">[6]</a> are using it. This means sequence snippets of length k from the database sequences are used to match against the reads to classify the read. Another approach is the Ferragina-Manzini (FM) index. This is a text-compression technique based on the Burrows-Wheeler transform. Centrifuge <a href="https://doi.org/10.1101%2Fgr.210641.116" title="Kim, D., Song, L., Breitwieser, F. P., &amp; Salzberg, S. L. (2016). Centrifuge: rapid and sensitive classification of metagenomic sequences. *Genome research*, 26(12), 1721-1729.">[7]</a>, for example, searches the reads in a forward and reverse manner, starting with an initial shorter match length which can be elongated if possible. Kaiju <a href="https://doi.org/10.1186/s13062-018-0208-7" title="Menzel, P., Ng, K. L., &amp; Krogh, A. (2016). Fast and sensitive taxonomic classification for metagenomics with Kaiju. *Nature communications*, 7(1), 1-9.">[8]</a>, too, uses the FM-index. <br>
Another difficulty of comparing the results of different classification tools are the various output formats. Some tools provide a summary report from which the identified species can be easily distinguished as well as the number of reads assigned to the different species or other taxonomic ranks. Other tools simply provide a list of read names and assign taxonomic IDs. Furthermore, the tools might have different taxonomic depths, some stop at the species rank and others even identify the strain. This complicates the comparison and usage within the same scripts.<br></p>
<p>This project aims to compare several classification tools regarding their usability for long-read data. The primarily used tools are Diamond <a href="https://doi.org/10.1038/nmeth.3176" title="Buchfink, B., Xie, C., &amp; Huson, D. H. (2015). Fast and sensitive protein alignment using DIAMOND. *Nature methods*, 12(1), 59-60.">[9]</a>, Kaiju, CCMetagen, Centrifuge, CLARK and Kraken2. For this comparison, common metrics like precision and recall are used to determine a single value, the Area under Precision-Recall Curve (AUPR). It provides a more realistic measure for the performance, since it considers various precision and recall values instead of just one raw value like the F1 score, for example <a href="https://doi.org/10.1016/j.cell.2019.07.010" title="Simon, H. Y., Siddle, K. J., Park, D. J., &amp; Sabeti, P. C. (2019). Benchmarking metagenomics tools for taxonomic classification. *Cell*, 178(4), 779-794.">[10]</a>. Another metric is the Abundance Profile Similarity (APS), which plays an important role for environmental metagenomic samples since the composition of a microbial population can give information about phenotypic effects <a href="https://doi.org/10.1016/j.cell.2019.07.010" title="Simon, H. Y., Siddle, K. J., Park, D. J., &amp; Sabeti, P. C. (2019). Benchmarking metagenomics tools for taxonomic classification. *Cell*, 178(4), 779-794.">[10]</a><a href="https://doi.org/10.1186/gb-2012-13-9-r79" title="Morgan, X. C., Tickle, T. L., Sokol, H., Gevers, D., Devaney, K. L., Ward, D. V., ... &amp; Huttenhower, C. (2012). Dysfunction of the intestinal microbiome in inflammatory bowel disease and treatment. *Genome Biology*, 13(9), 1-18.">[11]</a>. The last aspect to evaluate the performance of the chosen classifiers is the runtime. <br>
The tools are used for samples generated based on ZymoBIOMICS Microbial Community Standards and two kinds of databases: The first database is their default database, whereas the second database is based on the same RefSeq sequences for all tools. <br>
Additional to the comparison based on metrics, Multi Locus Sequence Typing (MLST) is used to validate the presence of the species in the dataset.</p>
<h1 id="material-and-methods">Material and Methods</h1>
<h2 id="data">Data</h2>
<p>The present dataset consists of four samples of the underlying ZymoBIOMICS Microbial Community Standards CS and CSII <a href="https://www.zymoresearch.de/collections/zymobiomics-microbial-community-standards/products/zymobiomics-microbial-community-standard" title="Zymo Research Corporation, Irvine, CA, USA. Product D6300, Lot ZRC190633">[12]</a><a href="https://www.zymoresearch.de/collections/zymobiomics-microbial-community-standards/products/zymobiomics-microbial-community-standard-ii-log-distribution" title="Zymo Research Corporation, Irvine, CA, USA. Product D6310, Lot ZRC190842">[13]</a><a href="https://doi.org/10.1093/gigascience/giz043" title="Nicholls, S. M., Quick, J. C., Tang, S., &amp; Loman, N. J. (2019). Ultra-deep, long-read nanopore sequencing of mock microbial community standards. *Gigascience*, 8(5), giz043.">[14]</a>. Those mock communities are composed of ten microbial species, eight bacteria and two fungi (<a href="../attachments/tables/table1.md" title="Table 1: General Information about the Used Species"><em><strong>Table 1</strong></em></a>). Each of these two standards is sequenced with GridION and PromethION, resulting in four samples, two for each standard.</p>
<table>
<thead>
<tr>
<th style="text-align:center">Name</th>
<th style="text-align:center">Type</th>
<th style="text-align:center">Taxonomic ID</th>
<th style="text-align:center">CS EVEN</th>
<th style="text-align:center"></th>
<th style="text-align:center">CS LOG</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"></td>
<td style="text-align:center"></td>
<td style="text-align:center"></td>
<td style="text-align:center">Expected Abundance (%)</td>
<td style="text-align:center">Estimated Abundance (%)</td>
<td style="text-align:center">Expected Abundance (%)</td>
</tr>
<tr>
<td style="text-align:center"><em>Bacillus subtilis</em></td>
<td style="text-align:center">Gram+</td>
<td style="text-align:center">1423</td>
<td style="text-align:center">12</td>
<td style="text-align:center">19.32 (19.02)</td>
<td style="text-align:center">0.89</td>
</tr>
<tr>
<td style="text-align:center"><em>Listeria monocytogenes</em></td>
<td style="text-align:center">Gram+</td>
<td style="text-align:center">1639</td>
<td style="text-align:center">12</td>
<td style="text-align:center">14.56 (14.33)</td>
<td style="text-align:center">89.1</td>
</tr>
<tr>
<td style="text-align:center"><em>Enterococcus faecalis</em></td>
<td style="text-align:center">Gram+</td>
<td style="text-align:center">1351</td>
<td style="text-align:center">12</td>
<td style="text-align:center">12.24 (12.07)</td>
<td style="text-align:center">0.00089</td>
</tr>
<tr>
<td style="text-align:center"><em>Staphylococcus aureus</em></td>
<td style="text-align:center">Gram+</td>
<td style="text-align:center">1280</td>
<td style="text-align:center">12</td>
<td style="text-align:center">11.28 (11.11)</td>
<td style="text-align:center">0.000089</td>
</tr>
<tr>
<td style="text-align:center"><em>Salmonella enterica</em></td>
<td style="text-align:center">Gram-</td>
<td style="text-align:center">28901</td>
<td style="text-align:center">12</td>
<td style="text-align:center">9.99 (10.32)</td>
<td style="text-align:center">0.089</td>
</tr>
<tr>
<td style="text-align:center"><em>Escherichia coli</em></td>
<td style="text-align:center">Gram-</td>
<td style="text-align:center">562</td>
<td style="text-align:center">12</td>
<td style="text-align:center">9.93 (10.26)</td>
<td style="text-align:center">0.089</td>
</tr>
<tr>
<td style="text-align:center"><em>Pseudomonas aeruginosa</em></td>
<td style="text-align:center">Gram-</td>
<td style="text-align:center">287</td>
<td style="text-align:center">12</td>
<td style="text-align:center">9.7 (10.11)</td>
<td style="text-align:center">8.9</td>
</tr>
<tr>
<td style="text-align:center"><em>Lactobacillus fermentum</em></td>
<td style="text-align:center">Gram+</td>
<td style="text-align:center">1613</td>
<td style="text-align:center">12</td>
<td style="text-align:center">9.28 (9.13)</td>
<td style="text-align:center">0.0089</td>
</tr>
<tr>
<td style="text-align:center"><em>Saccharomyces cerevisiae</em></td>
<td style="text-align:center">Yeast</td>
<td style="text-align:center">4932</td>
<td style="text-align:center">2</td>
<td style="text-align:center">1.92 (1.87)</td>
<td style="text-align:center">0.89</td>
</tr>
<tr>
<td style="text-align:center"><em>Cryptococcus neoformans</em></td>
<td style="text-align:center">Yeast</td>
<td style="text-align:center">5207</td>
<td style="text-align:center">2</td>
<td style="text-align:center">1.78 (1.77)</td>
<td style="text-align:center">0.00089</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table1.md" title="Table 1: General Information about the Used Species"><em><strong>Table 1: General Information about the Used Species.</strong></em></a> This table shows the different species included in the mock communities with their taxonomic ID and expected as well as estimated proportion of species. Note that the CS Even samples have different measured proportions for GridION (PromethION) than expected. There is no information about the measured proportions for CS Log. The information are gathered from <a href="https://www.zymoresearch.de/collections/zymobiomics-microbial-community-standards/products/zymobiomics-microbial-community-standard" title="Zymo Research Corporation, Irvine, CA, USA. Product D6300, Lot ZRC190633">[12]</a>, <a href="https://www.zymoresearch.de/collections/zymobiomics-microbial-community-standards/products/zymobiomics-microbial-community-standard-ii-log-distribution" title="Zymo Research Corporation, Irvine, CA, USA. Product D6310, Lot ZRC190842">[13]</a> and <a href="https://doi.org/10.1093/gigascience/giz043" title="Nicholls, S. M., Quick, J. C., Tang, S., &amp; Loman, N. J. (2019). Ultra-deep, long-read nanopore sequencing of mock microbial community standards. *Gigascience*, 8(5), giz043.">[14]</a>.<br> <br></p>
<table>
<thead>
<tr>
<th>FASTQ accession</th>
<th style="text-align:center">ERR3152364</th>
<th style="text-align:center">ERR3152366</th>
<th style="text-align:center">ERR3152365</th>
<th style="text-align:center">ERR3152367</th>
</tr>
</thead>
<tbody>
<tr>
<td>Sequencer</td>
<td style="text-align:center">GridION</td>
<td style="text-align:center">GridION</td>
<td style="text-align:center">PromethION</td>
<td style="text-align:center">PromethION</td>
</tr>
<tr>
<td>Zymo Community Standard</td>
<td style="text-align:center">CS Even</td>
<td style="text-align:center">CS Log</td>
<td style="text-align:center">CS Even</td>
<td style="text-align:center">CS Log</td>
</tr>
<tr>
<td>Used Name</td>
<td style="text-align:center">GridION364</td>
<td style="text-align:center">GridION366</td>
<td style="text-align:center">PromethION365</td>
<td style="text-align:center">PromethION367</td>
</tr>
<tr>
<td>Reads (M)</td>
<td style="text-align:center">3.59</td>
<td style="text-align:center">3.67</td>
<td style="text-align:center">35.7</td>
<td style="text-align:center">34.5</td>
</tr>
<tr>
<td>Quality (Median Q)</td>
<td style="text-align:center">10.3</td>
<td style="text-align:center">9.8</td>
<td style="text-align:center">10.5</td>
<td style="text-align:center">10.7</td>
</tr>
<tr>
<td>N50 (kb)</td>
<td style="text-align:center">5.3</td>
<td style="text-align:center">5.4</td>
<td style="text-align:center">5.4</td>
<td style="text-align:center">5.4</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table2.md" title="Table 2: General Information about the Samples"><em><strong>Table 2: General Information about the Samples.</strong></em></a> This table shows different information about the four samples of this project, including quality and read length (N50) as well as information about the used terms. In general, the PromethION samples are sequenced with greater depth resulting in more than 34 million reads, whereas the GridION samples contain around 3.6 million reads. The quality of all samples ranges from 9.8 to 10.7. The N50 is 5.4 except for GridION364, where it is 5.3. The information are gathered from Nicholls et al. in 2019 <a href="https://doi.org/10.1093/gigascience/giz043" title="Nicholls, S. M., Quick, J. C., Tang, S., &amp; Loman, N. J. (2019). Ultra-deep, long-read nanopore sequencing of mock microbial community standards. *Gigascience*, 8(5), giz043.">[14]</a>.<br> <br></p>
<p>The samples sequenced with PromethION show a higher read count with an average of 35.1 million reads and a median quality of 10.6. The GridION samples consist of approximately 3.5 million reads with a quality of 10 (<a href="../attachments/tables/table2.md" title="Table 2: General Information about the Samples"><em><strong>Table 2</strong></em></a>) <a href="https://doi.org/10.1093/gigascience/giz043" title="Nicholls, S. M., Quick, J. C., Tang, S., &amp; Loman, N. J. (2019). Ultra-deep, long-read nanopore sequencing of mock microbial community standards. *Gigascience*, 8(5), giz043.">[14]</a>.
The ZymoBIOMICS Microbial Community Standards come with knowledge about the abundance of the different species, which differ for CS and CSII. In the following, CS will be referred to as CS Even <a href="https://www.zymoresearch.de/collections/zymobiomics-microbial-community-standards/products/zymobiomics-microbial-community-standard" title="Zymo Research Corporation, Irvine, CA, USA. Product D6300, Lot ZRC190633">[12]</a>, since the abundances are 12% for each bacterial species and 2% for the two fungi, whereas CSII abundances follow a log distribution and will be referred to as CS Log <a href="https://www.zymoresearch.de/collections/zymobiomics-microbial-community-standards/products/zymobiomics-microbial-community-standard-ii-log-distribution" title="Zymo Research Corporation, Irvine, CA, USA. Product D6310, Lot ZRC190842">[13]</a>. The different abundances (expected and estimated) can be seen in <a href="../attachments/tables/table1.md" title="Table 1: General Information about the Used Species"><em><strong>Table 1</strong></em></a> <a href="https://www.zymoresearch.de/collections/zymobiomics-microbial-community-standards/products/zymobiomics-microbial-community-standard" title="Zymo Research Corporation, Irvine, CA, USA. Product D6300, Lot ZRC190633">[12]</a><a href="https://www.zymoresearch.de/collections/zymobiomics-microbial-community-standards/products/zymobiomics-microbial-community-standard-ii-log-distribution" title="Zymo Research Corporation, Irvine, CA, USA. Product D6310, Lot ZRC190842">[13]</a><a href="https://doi.org/10.1093/gigascience/giz043" title="Nicholls, S. M., Quick, J. C., Tang, S., &amp; Loman, N. J. (2019). Ultra-deep, long-read nanopore sequencing of mock microbial community standards. *Gigascience*, 8(5), giz043.">[14]</a>.</p>
<h2 id="tools">Tools</h2>
<p>The detailed commands for the classifcation and postprocessing of the outputs can be seen in the <a href="../Snakefile" title="Snakefile">Snakefile</a>, the detailed commands for database construction are saved in <a href="../Snakefile_DB" title="Snakefile for Database Creation">Snakefile_DB</a>. The used instructions to download the default databases or indices can be seen in <a href="../attachments/commands.md" title="Installation and Preparation, Information about Default Databases">here</a>.<br></p>
<p>Every tool is used on two different databases: the default database describes the databases or indices provided by the tool that can be downloaded. The custom database for each tool is based on the same sequences: the RefSeq sequences for bacteria and fungi downloaded from <a href="ftp://ftp.ncbi.nlm.nih.gov/" title="FTP Server">NCBI</a> as of 01/04/2021, see <a href="../attachments/paths.md" title="Line 1ff., 'Bacterial and Fungal RefSeq'">command</a>. For the protein tools, the bacterial proteome sequences present at this <a href="../attachments/paths.md" title="Line 4, 'Bacterial Proteome'">location</a> are used, the fungal sequences, stored <a href="../attachments/paths.md" title="Line 5, 'Fungal Proteome'">here</a> are downloaded from <a href="ftp://ftp.ncbi.nlm.nih.gov/" title="FTP Server">NCBI</a> as of 12/02/2021.</p>
<h3 id="classification">Classification</h3>
<table>
<thead>
<tr>
<th style="text-align:center">Tool</th>
<th style="text-align:center">Version</th>
<th style="text-align:center">Type</th>
<th style="text-align:center">Approach</th>
<th style="text-align:center">Default Database</th>
<th style="text-align:center">Creation Date</th>
<th style="text-align:left">Reference</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center">Diamond</td>
<td style="text-align:center">2.0.5</td>
<td style="text-align:center">Protein</td>
<td style="text-align:center">Alignment</td>
<td style="text-align:center">full proteome</td>
<td style="text-align:center">27/07/2019</td>
<td style="text-align:left">http://www.diamondsearch.org/index.php</td>
</tr>
<tr>
<td style="text-align:center">Kaiju</td>
<td style="text-align:center">1.7.4</td>
<td style="text-align:center">Protein</td>
<td style="text-align:center">FM-Index, Alignment</td>
<td style="text-align:center">RefSeq</td>
<td style="text-align:center">25/05/2020</td>
<td style="text-align:left">http://kaiju.binf.ku.dk/</td>
</tr>
<tr>
<td style="text-align:center">CCMetagen</td>
<td style="text-align:center">1.2.3</td>
<td style="text-align:center">DNA</td>
<td style="text-align:center"><em>k</em>-mer, Alignment (KMA)</td>
<td style="text-align:center">NCBI nt</td>
<td style="text-align:center">08/05/2019</td>
<td style="text-align:left">https://github.com/vrmarcelino/CCMetagen</td>
</tr>
<tr>
<td style="text-align:center">Centrifuge</td>
<td style="text-align:center">1.0.4</td>
<td style="text-align:center">DNA</td>
<td style="text-align:center">FM-Index</td>
<td style="text-align:center">RefSeq</td>
<td style="text-align:center">15/04/2018</td>
<td style="text-align:left">https://ccb.jhu.edu/software/centrifuge/manual.shtml</td>
</tr>
<tr>
<td style="text-align:center">CLARK</td>
<td style="text-align:center">1.2.5</td>
<td style="text-align:center">DNA</td>
<td style="text-align:center"><em>k</em>-mer</td>
<td style="text-align:center">RefSeq</td>
<td style="text-align:center">07/12/2020</td>
<td style="text-align:left">http://CLARK.cs.ucr.edu/Overview/</td>
</tr>
<tr>
<td style="text-align:center">Kraken2</td>
<td style="text-align:center">2.0.7-beta</td>
<td style="text-align:center">DNA</td>
<td style="text-align:center"><em>k</em>-mer</td>
<td style="text-align:center">-</td>
<td style="text-align:center">2019/2020</td>
<td style="text-align:left">http://ccb.jhu.edu/software/kraken2/</td>
</tr>
<tr>
<td style="text-align:center">BugSeq</td>
<td style="text-align:center">v1</td>
<td style="text-align:center">DNA</td>
<td style="text-align:center">Pipeline</td>
<td style="text-align:center">RefSeq</td>
<td style="text-align:center">-</td>
<td style="text-align:left">https://bugseq.com/free</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table3.md" title="Table 3: Overview of Used Classification Tools and General Information"><em><strong>Table 3: Overview of Used Classification Tools and General Information.</strong></em></a> The creation dates are based on the date the donwloaded databases or indices are generated. For Kraken2, no exact can be given, since the <a href="../attachments/paths.md" title="Line 11, 'Kraken2 Default Database'">files</a> have different creation or alteration dates. For BugSeq, no date can be given either. The database was generated 23/02/2020 and since then, updated each month <a href="https://doi.org/10.1186/s12859-021-04089-5" title="Fan, J., Huang, S., &amp; Chorlton, S. D. (2021). BugSeq: a highly accurate cloud platform for long-read metagenomic analyses. *BMC bioinformatics*, 22(1), 1-12.">[25]</a>.</p>
<h4 id="diamond">Diamond</h4>
<p>This tool is a sequence aligner for protein and translated DNA searches specifically designed for big sequence data. The approach for alignments and classification is called seed-and-extend, i.e. short exact matches between query and reference sequence are searched and elongated if possible <a href="https://doi.org/10.1038/nmeth.3176" title="Buchfink, B., Xie, C., &amp; Huson, D. H. (2015). Fast and sensitive protein alignment using DIAMOND. *Nature methods*, 12(1), 59-60.">[9]</a>. <br>
The default database is based on the <a href="../attachments/paths.md" title="Line 4, 'Bacterial Proteome'">full bacterial proteome</a> (as of 27/07/2019), whereas the custom database consists of the beforementioned RefSeq sequences of bacterial and fungal proteome.</p>
<pre><code>diamond makedb --in {input.faa} -d prefix --taxonmap {map} --taxonnodes {nodes.dmp} --taxonnames {names.dmp}

# Parameters:
#   -d      prefix of database file name
</code></pre>
<p>To use Diamond for taxonomic classification, the output format has to be defined. This is done with <tt>--outfmt 102</tt>. For the custom database and the classification of GridION366, the block size is decreased to 1.0, 0.5 and 0.25, it considers the RAM usage.</p>
<pre><code>diamond blastx --db {database} -q {input.fastq} -o {output} -p {threads} --outfmt 102 [-b1.0 -t {tmp_dir}]

# Parameters:
#   --outfmt    output format, 102 for taxonomic classification (default: BLAST tabular format)
#   -b          sequence block size in billions of letters (default: 2.0)    
#   -t          folder for temporary files    
</code></pre>
<h4 id="kaiju">Kaiju</h4>
<p>Kaiju is a fast and sensitive tool for taxonomic classification of metagenomic samples that uses a reference database of annotated protein-coding genes of microbial genomes. The DNA reads are translated into the six reading frames and split according to the stop codons. The resulting fragments are sorted regarding their length and due to the usage of the Ferragina and Manzini-Index (FM-Index), exact matches can be searched between read and database reference in efficient time by Kaiju <a href="https://doi.org/10.1186/s13062-018-0208-7" title="Menzel, P., Ng, K. L., &amp; Krogh, A. (2016). Fast and sensitive taxonomic classification for metagenomics with Kaiju. *Nature communications*, 7(1), 1-9.">[8]</a>. <br>
The default database consists of the downloaded index <a href="../attachments/paths.md" title="Line 6, 'Kaiju Default Index'"><tt>kaiju_db_refseq_2020-05-25</tt></a> which was downloaded from the <a href="http://kaiju.binf.ku.dk/server">Kaiju webserver</a> as of 28/11/2020. <br>
Building a custom database consists of two steps. The sequence headers need to consist of the NCBI taxonomic ID and optionally a prefix separated with an underscore (e.g. <tt>&gt;2_1423</tt>). To achieve this format, the protein sequence file <a href="../attachments/paths.md" title="Line 7, 'RefSeq Proteome Bacteria and Fungi'">refseq_bac_fung.faa</a> is altered with the script <a href="scripts/../../scripts/changeHeaderKaiju.py">changeHeaderKaiju.py</a>, which is based on a script of <a href="https://github.com/vrmarcelino/CCMetagen/blob/master/benchmarking/rename_nt/rename_nt.py" title="Renaming script from CCMetagen">CCMetagen</a>. The commands to build the custom database can be seen in the following box.</p>
<pre><code>kaiju-mkbwt -n {threads} -a ACDEFGHIKLMNPQRSTVWY -o {bwt} {input.faa}
kaiju-mkfmi -r {bwt}

# Parameters:
#   -a      used alphabet (basic proteins)
#   -r      removing unused files
#   -o      name of the BWT output file
</code></pre>
<p><tt>kaiju-mkbwt</tt> takes a fasta file as an input and generates the corresponding Burrows-Wheeler-Transform (BWT), whereas <tt>kaiju-mkfmi</tt> uses the BWT as an input to generate the needed index file. <br></p>
<p>To generate reports, the following command is used. Since this comparison is based on species level, the report is generated for species, hence the parameter <tt>-r species</tt>.</p>
<pre><code>kaiju -t {nodes.dmp} -f {index.fmi} -i {input.fastq} -o {output.classification}
kaiju2table -t {nodes.dmp} -n {names.dmp} -r species -o {output.report} {input.classification}

# Parameters
#   -r      taxonomic rank
</code></pre>
<h4 id="ccmetagen">CCMetagen</h4>
<p>The classifier CCMetagen <a href="https://doi.org/10.1186/s13059-020-02014-2" title="Marcelino, V. R., Clausen, P. T., Buchmann, J. P., Wille, M., Iredell, J. R., Meyer, W., ... &amp; Holmes, E. C. (2020). CCMetagen: comprehensive and accurate identification of eukaryotes and prokaryotes in metagenomic data. *Genome biology*, 21, 1-15.">[6]</a> uses a k-mer alignment (KMA) <a href="https://doi.org/10.1186/s12859-018-2336-6" title="Clausen, P. T., Aarestrup, F. M., &amp; Lund, O. (2018). Rapid and precise alignment of raw reads against redundant databases with KMA. *BMC bioinformatics*, 19(1), 1-8.">[15]</a> mapping method and produces ranked taxonomic classification results.<br>
The used <a href="../attachments/paths.md" title="Line 8, 'CCMetagen Default Index'">indexed database</a> was downloaded from their <a href="http://www.cbs.dtu.dk/public/CGE/databases/CCMetagen/">website</a> as of 08/12/2020. The preprocessing with KMA uses the following command, which is orientated on the suggestion of CCMetagen.</p>
<pre><code>kma -i {input} -t_db {database} -o {output.result} -t {threads} -1t1 -mem_mode -and -ef

# Parameters    
    # -1t1      no splicing, one read one template
    # -ef       extended output file that includes percentage of classified reads
    # -mem_mode files are not loaded into memory

CCMetagen.py -o result -i input.kma -ef y --map input.mapstat

# Parameters    
    # -i        path to kma result
    # -ef       extended output file that includes percentage of classified reads
</code></pre>
<p>The custom database using the RefSeq of bacteria and fungi needs to be created with KMA. The input sequence headers need to be altered to fit the necessary format. This is done with <a href="../scripts/changeHeaderCCMetagen.py">changeHeaderCCMetagen.py</a> which is based on the CCMetagen script for renaming nt sequenes (<a href="https://github.com/vrmarcelino/CCMetagen/blob/master/benchmarking/rename_nt/rename_nt.py" title="Renaming script from CCMetagen">CCMetagen script</a>). The parameter for a sparse database is used due to memory limits.</p>
<pre><code>kma_index -i {input} -o {output} -Sparse

# Parameters
    # -Sparse   for making a sparse DB
</code></pre>
<h4 id="centrifuge">Centrifuge</h4>
<p>Centrifuge <a href="https://doi.org/10.1101%2Fgr.210641.116" title="Kim, D., Song, L., Breitwieser, F. P., &amp; Salzberg, S. L. (2016). Centrifuge: rapid and sensitive classification of metagenomic sequences. *Genome research*, 26(12), 1721-1729.">[7]</a> is a classification tool for metagenomic microbial data. This FM-index-based approach searches for forward and reverse complements of the given input reads in the corresponding database of species. If a match with a given seed length is found, that region is expanded until a mismatch is found. <br>
The <a href="../attachments/paths.md" title="Line 9, 'Centrifuge Default Index'">used indices</a> are downloaded from their <a href="https://genome-idx.s3.amazonaws.com/centrifuge/">website</a> as of 15/01/2021. <br>
The custom database can be created with the following command, where <tt>prefix</tt> is the prefix the index files will carry. The <tt>seqid2taxid.map</tt> file generated from Kraken2 during database building is used. This spares the download of all sequences that should be included in the index.</p>
<pre><code>centrifuge-build -p {threads} --conversion-table {seqid2taxid.map} --taxonomy-tree {nodes.dmp} --name-table {names.dmp} {input.fna} {prefix}
</code></pre>
<p>For classification with Centrifuge, the default parameters are used which can be seen in the following command box. The parameter <tt>ignore-quals</tt> classifies with assuming the quality of each read is 30. <!-- The minimum hit length is set to the median read length of each sample.--></p>
<pre><code>centrifuge -q -x {index} {input.fastq} --report-file {report} -S {output} [--ignore-quals]
    
# Parameters:
    # -q                files are fastq
    # --ignore-quals    treat all quality values as 30 on Phred scale
</code></pre>
<!-- # --min-hit-length  minimum length of partial hits -->
<h4 id="clark">CLARK</h4>
<p>CLARK <a href="https://doi.org/10.1186/s12864-015-1419-2" title="Ounit, R., Wanamaker, S., Close, T. J., &amp; Lonardi, S. (2015). CLARK: fast and accurate classification of metagenomic and genomic sequences using discriminative k-mers. *BMC genomics*, 16(1), 1-13.">[5]</a> is a taxonomic classification tool for metagenomic samples of any format (reads, contigs, scaffolds, assemblies, ...). The method is based on discriminative k-mers which is used in supervised sequence classification. The bacterial (and virus) <a href="../attachments/paths.md" title="Line 10, 'CLARK Default Database'">default database</a> is build using the in-build script <tt>set_targets.sh</tt> (version 1.2.5) as of 7/12/2020. The custom database is build using the same script but custom sequences, however, the script is from version 2.1.6.1. Classification is done with the following command and, as mentioned in <a href="../attachments/tables/table3.md" title="Table 3: Overview of Used Classification Tools and General Information"><em><strong>Table 3</strong></em></a>, with CLARK's version 1.2.5.</p>
<pre><code>CLARK --long -O {input.fastq} -R {output} -D {database} -n {threads} -T {targets}
CLARK-l --long -O {input.fastq} -R {output} -D {database} -n {threads} -T {targets}
</code></pre>
<p>For classifying the samples with the custom database, CLARK-l is used, which works with limited memory. The rest of the command remains unchanged. However, the created index has k-mers of length 27 nt, instead of the default 31-mers.
Another in-build script <a href="../scripts/clark.estimate_abundance.sh" title="clark.estimate_abundance.sh"><tt>estimate_abundance.sh</tt></a> is used with alterations for generating a report file.<!-- Therefore, the command for CLARK-l is used for the classifications with the custom database. --></p>
<h4 id="kraken2">Kraken2</h4>
<p>The taxonomic sequence classifier Kraken2 examines k-mers of a query sequence and uses those information to query a database. During the query, the k-mers are mapped to the lowest common ancestor of the genomes that contain a given k-mer <a href="https://doi.org/10.1186/s13059-019-1891-0" title="Wood, D. E., Lu, J., &amp; Langmead, B. (2019). Improved metagenomic analysis with Kraken 2. *Genome biology*, 20(1), 1-13.">[4]</a>. <br>
The used default database is located <a href="../attachments/paths.md" title="Line 11, 'Kraken2 Default Database'">here</a>. The custom database can be generated using the following commands with the <tt>clean</tt> command removing unnecessary files. The used scripts for this build are version 2.1.1, the classification, however, is performed with version 2.0.7-beta (<a href="../attachments/tables/table3.md" title="Table 3: Overview of Used Classification Tools and General Information"><em><strong>Table 3</strong></em></a>).</p>
<pre><code>kraken2-build --download-taxonomy --db {database}
kraken2-build --add-to-library {input.fungi.fna} --db {database} --threads {threads}
kraken2-build --add-to-library {input.bacteria.fna} --db {database} --threads {threads}
kraken2-build --build {database} --threads {threads}
kraken2-build --clean
</code></pre>
<p>For taxonomic classification, the following command is used.</p>
<pre><code>kraken2 --db {database} --report {report} --threads {threads} --output {output} {input.fastq}
</code></pre>
<h4 id="bugseq">BugSeq</h4>
<p>BugSeq <a href="https://doi.org/10.1186/s12859-021-04089-5" title="Fan, J., Huang, S., &amp; Chorlton, S. D. (2021). BugSeq: a highly accurate cloud platform for long-read metagenomic analyses. *BMC bioinformatics*, 22(1), 1-12.">[25]</a> is a cloud-based pipeline for classification, therefore no parameters can be specified. The website can be reached with <a href="https://bugseq.com/free">https://bugseq.com/free</a>. However, the implemented tools are used with default parameters. <br>
First, quality control with fastp is performed, the reads are then mapped with minimap2 against RefSeq sequences of the corresponding database. Pathoscope reassigns the alignments to the reference sequences using a Bayesian statistical framework. The reads are classified based on the lowest common ancestor <a href="https://doi.org/10.1186/s12859-021-04089-5" title="Fan, J., Huang, S., &amp; Chorlton, S. D. (2021). BugSeq: a highly accurate cloud platform for long-read metagenomic analyses. *BMC bioinformatics*, 22(1), 1-12.">[25]</a>.  <br>
The advanced options consist of the choice of the metagenomic database. The options are either the 'BugSeq Default', which includes all bacterial genomes in RefSeq as well as fungal, protozoal and viral genomes in RefSeq, the human genome and transcriptome are included as well. The completion status is not taken into account. The other database is the NCBI nt, but for this analysis, it is not chosen but the 'BugSeq Default' database. <br>
The other option is a specification of the sample type. Since this dataset includes two fungal sequences, the option <tt>bacterial isolates</tt> could not be used, therefore the default <tt>Generic</tt> sample type has been chosen. The tool is available in two options: academic and lab. The academic version allows only for uploads up to 10 Gb, whereas the Lab version involves monthly payment, but has unlimited size. <!--It is currently not possible to try out the Lab version if there already exists an account. This means, no samples could be submitted since the smallest sample has a size of at least 11 Gb.--> <br></p>
<h3 id="others">Others</h3>
<p>Additional to the classification tools, conda <a href="https://docs.anaconda.com/" title="Anaconda Software Distribution. (2020). Anaconda Documentation. *Anaconda Inc.* Retrieved from https://docs.anaconda.com/">[16]</a> is used to organise and coordinate the different requirements of the tools. The tools themselves and their execution are structured with snakemake <a href="https://snakemake.readthedocs.io/en/stable/" title="Köster, J., &amp; Rahmann, S. (2012). Snakemake—a scalable bioinformatics workflow engine. Bioinformatics, 28(19), 2520-2522.">[17]</a>. Some analysis is done with Python, R and Bash (<a href="../attachments/tables/table4.md" title="Table 4: Overview of additional Software and the Used Versions"><em><strong>Table 4</strong></em></a>).</p>
<table>
<thead>
<tr>
<th></th>
<th>Version</th>
<th>Reference</th>
</tr>
</thead>
<tbody>
<tr>
<td>conda</td>
<td>4.7.5</td>
<td><a href="https://docs.anaconda.com/">conda</a></td>
</tr>
<tr>
<td>snakemake</td>
<td>3.10.0</td>
<td><a href="https://snakemake.readthedocs.io/en/stable/">snakemake</a></td>
</tr>
<tr>
<td>Python</td>
<td>3.8.8</td>
<td><a href="https://www.Python.org">Python</a></td>
</tr>
<tr>
<td>R</td>
<td>3.4.4</td>
<td><a href="https://www.r-project.org/">R</a></td>
</tr>
<tr>
<td>Bash</td>
<td>4.4.12(1)</td>
<td></td>
</tr>
<tr>
<td>makeblastdb</td>
<td>2.5.0+</td>
<td>https://blast.ncbi.nlm.nih.gov/Blast.cgi</td>
</tr>
<tr>
<td>blastn</td>
<td>2.7.1+</td>
<td>https://blast.ncbi.nlm.nih.gov/Blast.cgi</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table4.md" title="Table 4: Overview of additional Software and the Used Versions"><em><strong>Table 4: Overview of additional Software and the Used Versions.</strong></em></a> The usage of BLAST <a href="https://doi.org/10.1016/S0022-2836(05)80360-2" title="Altschul, S. F., Gish, W., Miller, W., Myers, E. W., &amp; Lipman, D. J. (1990). Basic local alignment search tool. *Journal of molecular biology*, 215(3), 403-410.">[18]</a> is explained in more detailed in the section of <a href="#multi-locus-sequence-typing">Multi Locus Sequence Typing</a>. <br> <br></p>
<h2 id="metrics">Metrics</h2>
<p>There is a difficulty in benchmarking different classifiers since the chosen metrics can affect the evaluated performances. <br>
Precision and recall are considered as the most important metrics in this context which are commonly used in benchmarking studies <a href="https://doi.org/10.1016/j.cell.2019.07.010" title="Simon, H. Y., Siddle, K. J., Park, D. J., &amp; Sabeti, P. C. (2019). Benchmarking metagenomics tools for taxonomic classification. *Cell*, 178(4), 779-794.">[10]</a>. The F1 score (harmonic mean of precision and recall) does not provide a realistic estimate of performance since it is based on a single precision and recall <a href="https://doi.org/10.1016/j.cell.2019.07.010" title="Simon, H. Y., Siddle, K. J., Park, D. J., &amp; Sabeti, P. C. (2019). Benchmarking metagenomics tools for taxonomic classification. *Cell*, 178(4), 779-794.">[10]</a>. <br></p>
<p>Dealing with a lot of different tools involves dealing with a lot of different output formats, therefore all the following calculations are based on preprocessed output formats <tt>areport</tt>. The Python scripts generating these outputs are specific for the different tools and can be found in <a href="../scripts">scripts</a> with the names <tt>(tool)Output.py</tt>. The areports consist of five columns: abundance, read count, taxonomic rank, taxonomic ID and scientific species name. For the analyses, only the entries on species level are considered. <br>
The abundances for most tools are calculated by dividing the number of assigned reads of a species by the total number of reads. Some tools, however, return more than one hit per read, therefore the abundances are calculated by dividing the number of assigned reads of that entry by the total number of assignments.</p>
<h3 id="area-under-precision-recall-curve">Area Under Precision-Recall Curve</h3>
<p>The Area Under Precision-Recall Curve (AUPR) is a metric that combines those most important measures for metagenomic classification: precision and recall <a href="https://doi.org/10.1016/j.cell.2019.07.010" title="Simon, H. Y., Siddle, K. J., Park, D. J., &amp; Sabeti, P. C. (2019). Benchmarking metagenomics tools for taxonomic classification. *Cell*, 178(4), 779-794.">[10]</a>. Precision is defined as $\frac{TP}{TP+FP}$, i.e. the ratio between true positive (TP) classification results and the total number of classification results that are reported as true, including false positive (FP) hits. Recall, or sensitivity, on the other hand, is defined as the ratio of true positives against all correct classifications including false negatives (FN), i.e. $\frac{TP}{TP+FN}$. <!-- recall <sup>TP</sup>&frasl;<sub>TP+FN</sub> $precision=\frac{TP}{TP+FP}$ $recall=\frac{TP}{TP+FN}$ precisio n<sup>TP</sup>&frasl;<sub>TP+FP</sub>--> <br>
The AUPR provides a more realistic measure for the performance of classifiers since it is not based on one raw value (precision, recall, F1, ...) but instead on a list of values for different thresholds <a href="https://doi.org/10.1016/j.cell.2019.07.010" title="Simon, H. Y., Siddle, K. J., Park, D. J., &amp; Sabeti, P. C. (2019). Benchmarking metagenomics tools for taxonomic classification. *Cell*, 178(4), 779-794.">[10]</a>. <br>
The AUPR curve can be used to evaluate precision and recall across different abundance thresholds. If the thresholds are chosen accordingly in the range from 0-1.0, AUPR returns a single metric considering precision and recall. In short, this metric considers the number of correctly identified species <a href="https://doi.org/10.1016/j.cell.2019.07.010" title="Simon, H. Y., Siddle, K. J., Park, D. J., &amp; Sabeti, P. C. (2019). Benchmarking metagenomics tools for taxonomic classification. *Cell*, 178(4), 779-794.">[10]</a>. To identify TP and FP hits, a ground truth is needed, which is given due to the underlying ZymoBIOMICS Microbial Community Standards. Only the entries at species rank are considered.<br>
The Precision-Recall Curves and the AUPR are calculated within the bash script <a href="../scripts/PRCurve.sh">PRCurve.sh</a> which uses the Python script <a href="../scripts/extractingVal4Vis.py">extractingVal4Vis.py</a> for extracting the ground truth vectors and vectors with abundances of the species as an input for the R script <a href="../scripts/visPRCurve.R">visPRCurve.R</a> which then visualizes the values as a precision-recall curve. The used R package is <tt>PRROC</tt> <a href="https://cran.r-project.org/web/packages/PRROC/index.html" title="Jens Keilwagen, Ivo Grosse and Jan Grau (2014). Area under Precision-Recall Curves for Weighted and Unweighted Data. *PLOS ONE* (9) 3.">[19]</a>. <br>
The resulting plots also contain a baseline, which is different for each sample and represents the lowest precision. It is the number of true positives divided by the number of all hits.</p>
<h3 id="abundance-profile-similarity">Abundance Profile Similarity</h3>
<p>This metric is based on the abundances the different classifiers detect for the given species of the sample. This might be important as the changes in microbial population composition play a role in phenotypic effect <a href="https://doi.org/10.1016/j.cell.2019.07.010" title="Simon, H. Y., Siddle, K. J., Park, D. J., &amp; Sabeti, P. C. (2019). Benchmarking metagenomics tools for taxonomic classification. *Cell*, 178(4), 779-794.">[10]</a><a href="https://doi.org/10.1186/gb-2012-13-9-r79" title="Morgan, X. C., Tickle, T. L., Sokol, H., Gevers, D., Devaney, K. L., Ward, D. V., ... &amp; Huttenhower, C. (2012). Dysfunction of the intestinal microbiome in inflammatory bowel disease and treatment. *Genome Biology*, 13(9), 1-18.">[11]</a>. The abundance that is considered here, is based on the number of reads assigned to a certain species divided by the total number of reads/assigned reads, therefore the read counts are not corrected for genome size. <br>
The abundance profiles are determined with the pairwise distances between the abundances of ground truth and the abundances estimated by the classifiers at the species level, i.e. the L2 distance <a href="https://doi.org/10.1016/j.cell.2019.07.010" title="Simon, H. Y., Siddle, K. J., Park, D. J., &amp; Sabeti, P. C. (2019). Benchmarking metagenomics tools for taxonomic classification. *Cell*, 178(4), 779-794.">[10]</a>. To calculate the euclidian distance, the Python script <a href="../scripts/abundanceProfileSimilarity.py">abundanceProfileSimilarity.py</a> is used which uses Scipy's <a href="https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.euclidean.html" title="SciPy manual, Spatial Distance"><tt>spatial.distance</tt></a> package.</p>
<h3 id="computational-requirements">Computational Requirements</h3>
<p>Additionally to the quality of the different classifiers, the computational requirements are compared, i.e. the runtime for classification and database construction, if possible. They are measured using the <tt>benchmark</tt> option in <tt>snakemake</tt>, which returns the wall clock time of a task. <br>
The runs are performed on <tt>Linux prost 4.9.0-13-amd64 #1 SMP Debian 4.9.228-1 (2020-07-05) x86_64 GNU/Linux</tt> where eight threads are used for classification and 16 for database creation.</p>
<!-- The results are also briefly compared to Simon H. Ye et al. [[10]](https://doi.org/10.1016/j.cell.2019.07.010 "Simon, H. Y., Siddle, K. J., Park, D. J., & Sabeti, P. C. (2019). Benchmarking metagenomics tools for taxonomic classification. *Cell*, 178(4), 779-794."). -->
<h2 id="multi-locus-sequence-typing">Multi Locus Sequence Typing</h2>
<p>Multi Locus Sequence Typing (MLST) describes using, in general, seven preserved and well-known genes or loci to identify species and especially strains and therefore to analyse the molecular evolution <a href="https://www.sciencedirect.com/science/article/pii/S0580951715000148" title="Dingle, T. C., &amp; MacCannell, D. R. (2015). Molecular strain typing and characterisation of toxigenic Clostridium difficile. *Methods in Microbiology*, 42, 329-357.">[20]</a>. From the genes, internal fragments of 450 to 500 bp are used. There are different sequences for each gene representing the different alleles. A selection of those alleles for each strain/isolate defines the sequence type (ST) or allelic profile. The collection of ST is the MLST scheme for the species, i.e. each isolate/strain is characterized by the alleles at the gene loci <a href="https://pubmlst.org/multilocus-sequence-typing" title="Public databases for molecular typing and microbial genome diversity. Multi-Locus Sequence Typing. Retrieved from https://pubmlst.org/multilocus-sequence-typing. Last visited on 21/04/2021.">[21]</a>. <br>
Here, MLSTs are used as an additional validation of the performances of the classifiers. <br></p>
<p>The chosen sequence types for each species are retrieved from <a href="https://pubmlst.org/">PubMLST</a> for most species. An overview of the used genes and alleles can be seen in <a href="../attachments/tables/table5.md" title="Table 5: Overview of MLST schemes for Given Species"><em><strong>Table 5</strong></em></a>. The first ST is selected for each species. <br>
The database has no entry for <em>Lactobacillus fermentum</em>, therefore the MLST is retrieved from Poluektova et al. <a href="https://doi.org/10.1007/s00203-017-1346-5" title="Poluektova, E. U., Yunes, R. A., Epiphanova, M. V., Orlova, V. S., &amp; Danilenko, V. N. (2017). The Lactobacillus rhamnosus and Lactobacillus fermentum strains from human biotopes characterized by MLST and toxin-antitoxin gene polymorphism. *Archives of microbiology*, 199(5), 683-690.">[22]</a>. The first entry for each gene is used. <br>
The MLST housekeeping-genes for <em>Cryptococcus neoformans</em> are based on Meyer et al. <a href="https://doi.org/10.1080/13693780902953886" title="Meyer, W., Aanensen, D. M., Boekhout, T., Cogliati, M., Diaz, M. R., Esposto, M. C., ... &amp; Kwon-Chung, J. (2009). Consensus multi-locus sequence typing scheme for Cryptococcus neoformans and Cryptococcus gattii. *Medical mycology*, 47(6), 561-570.">[23]</a>, the sequences are gathered from the <a href="https://mlst.mycologylab.org/">International Fungal Multi Locus sequence Typing Database</a>. For the allele type number, 1 is chosen across all genes. <br>
The sequences chosen for <em>Saccharomyces cerevisiae</em> are based on Eeom, YJ., Son, SY., Jung, DH. et al. <a href="https://doi.org/10.1007/s10068-018-0335-z" title="Eeom, Y. J., Son, S. Y., Jung, D. H., Hur, M. S., Kim, C. M., Park, S. Y., ... &amp; Park, C. S. (2018). Diversity analysis of Saccharomyces cerevisiae isolated from natural sources by multilocus sequence typing (MLST). *Food science and biotechnology*, 27(4), 1119-1127.">[24]</a> and gathered from <a href="https://www.ncbi.nlm.nih.gov/genbank/">GenBank</a>. The specific files can be seen <a href="../attachments/paths.md" title="Lines 13ff, 'MLST Saccharomyces Cerevisiae'">here</a>. The sequence primers are searched for in the corresponding files and depending on the discovery of forward or reverse primers, the specified number of nucleotides in front or following the primer are chosen as gene fragments.</p>
<table>
<thead>
<tr>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="https://pubmlst.org/bigsdb?db=pubmlst_bsubtilis_seqdef&amp;page=downloadProfiles&amp;scheme_id=1" title="MLST scheme, B. subtilis"><em>Bacillus subtilis</em></a></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>Escherichia coli</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>glpF</td>
<td>ilvD</td>
<td>pta</td>
<td>purH</td>
<td>pycA</td>
<td>rpoD</td>
<td>tpiA</td>
<td></td>
<td></td>
<td>dinB</td>
<td>icdA</td>
<td>pabB</td>
<td>polB</td>
<td>putP</td>
<td>trpA</td>
<td>trpB</td>
<td>uidA</td>
</tr>
<tr>
<td></td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td></td>
<td></td>
<td>1</td>
<td>1</td>
<td>2</td>
<td>1</td>
<td>1</td>
<td>2</td>
<td>3</td>
<td>1</td>
</tr>
<tr>
<td><a href="https://bigsdb.pasteur.fr/cgi-bin/bigsdb/bigsdb.pl?db=pubmlst_listeria_seqdef&amp;page=downloadProfiles&amp;scheme_id=2" title="MLST scheme, L. monocytogenes"><em>Listeria monocytogenes</em></a></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td><a href="https://pubmlst.org/bigsdb?db=pubmlst_paeruginosa_seqdef&amp;page=downloadProfiles&amp;scheme_id=1" title="MLST scheme, P. aeruginosa"><em>Pseudomonas aeruginosa</em></a></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>abcZ</td>
<td>bglA</td>
<td>cat</td>
<td>dapE</td>
<td>dat</td>
<td>ldh</td>
<td>lhkA</td>
<td></td>
<td></td>
<td>acsA</td>
<td>aroE</td>
<td>guaA</td>
<td>mutL</td>
<td>nuoD</td>
<td>ppsA</td>
<td>trpE</td>
<td></td>
</tr>
<tr>
<td></td>
<td>3</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>3</td>
<td>1</td>
<td>3</td>
<td></td>
<td></td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td></td>
</tr>
<tr>
<td><a href="https://pubmlst.org/bigsdb?db=pubmlst_efaecalis_seqdef&amp;page=downloadProfiles&amp;scheme_id=1" title="MLST scheme, E. faecalis"><em>Enterococcus faecalis</em></a></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td><a href="https://static-content.springer.com/esm/art%3A10.1007%2Fs00203-017-1346-5/MediaObjects/203_2017_1346_MOESM1_ESM.docx" title="MLST from Supplements [22], L. fermentum"><em>Lactobacillus fermentum</em></a></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>gdh</td>
<td>gyd</td>
<td>pstS</td>
<td>gki</td>
<td>aroE</td>
<td>xpt</td>
<td>yqiL</td>
<td></td>
<td></td>
<td>parB</td>
<td>ychF</td>
<td>pyrG</td>
<td>atpF</td>
<td>recA</td>
<td>ileS</td>
<td>recG</td>
<td>leuS</td>
</tr>
<tr>
<td></td>
<td>3</td>
<td>1</td>
<td>16</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td></td>
<td></td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
</tr>
<tr>
<td><a href="https://pubmlst.org/bigsdb?db=pubmlst_saureus_seqdef&amp;page=downloadProfiles&amp;scheme_id=1" title="MLST scheme, S. aureus"><em>Staphylococcus aurueus</em></a></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td><a href="https://doi.org/10.1007/s10068-018-0335-z" title="Eeom, Y. J., Son, S. Y., Jung, D. H., Hur, M. S., Kim, C. M., Park, S. Y., ... &amp; Park, C. S. (2018). Diversity analysis of Saccharomyces cerevisiae isolated from natural sources by multilocus sequence typing (MLST). *Food science and biotechnology*, 27(4), 1119-1127.">Saccharomyces cerevisiae, [24]</a></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>arcC</td>
<td>aroE</td>
<td>glpF</td>
<td>gmk</td>
<td>pta</td>
<td>tpi</td>
<td>yqiL</td>
<td></td>
<td></td>
<td>acc1</td>
<td>gln4</td>
<td>adp1</td>
<td>rpn2</td>
<td>meta4</td>
<td>nup116</td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td></td>
<td></td>
<td>r</td>
<td>f</td>
<td>r</td>
<td>r</td>
<td>f</td>
<td>r</td>
<td></td>
<td></td>
</tr>
<tr>
<td><a href="https://pubmlst.org/bigsdb?db=pubmlst_mlst_seqdef&amp;page=downloadProfiles&amp;scheme_id=2" title="MLST scheme, S. enterica"><em>Salmonella enterica</em></a></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td><a href="https://mlst.mycologylab.org/page/CN%20Main" title="Database, C. neoformans"><em>Cryptococcus neoformans</em></a></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>aroC</td>
<td>dnaN</td>
<td>hemD</td>
<td>hisD</td>
<td>purE</td>
<td>sucA</td>
<td>thrA</td>
<td></td>
<td></td>
<td>cap59</td>
<td>gpd1</td>
<td>lac1</td>
<td>plb1</td>
<td>sod1</td>
<td>ura5</td>
<td>lgs1</td>
<td></td>
</tr>
<tr>
<td></td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>5</td>
<td></td>
<td></td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td>1</td>
<td></td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table5.md" title="Table 5: Overview of MLST schemes for Given Species"><em><strong>Table 5: Overview of MLST schemes for Given Species.</strong></em></a> This table shows the selected gene fragments for each species. The first line contains the included genes whereas the second line states which locus is used in that allelic profile. For <em>S. cerevisiae</em>, it is noted which primer is used as an anchor point for choosing the gene fragment. The used MLST schemes or origins from where the sequences are gatheres are linked with the species name. <br> <br></p>
<p>The sequences are concatenated into the file <a href="../attachments/paths.md" title="Line 12, 'MLST'">MLSTS.fasta</a> and blasted against each sample. To generate a database for each sample, the following command is used.</p>
<pre><code>makeblastdb -in input.fasta -parse_seqids -dbtype nucl -out output.prefix
</code></pre>
<p>Since the use of those MLSTs is to validate the occurence of species in the sample, it is sufficient to return one hit per sequene in the query file. This is done by setting the parameter <tt>max_targets_seq</tt> and <tt>culling_limit</tt>. This leads to the following command for blasting.</p>
<pre><code>blastn -task blastn -outfmt 6 -max_target_seqs 1 -culling_limit 1 -query MLST.fasta -db sample.db -out sample.blast
</code></pre>
<h1 id="results-and-discussion">Results and Discussion</h1>
<p>The following results are based on the <tt>areport</tt> format introduced before. The files can be seen <a href="../areports" title="Folder for areports">here</a>. The classification results for each tool can be found <a href="../attachments/paths.md" title="Line 29, Results Classification">here</a>.</p>
<h2 id="comparison-using-the-metrics">Comparison Using the Metrics</h2>
<h3 id="benchmarking-with-area-under-precision-recall-curve">Benchmarking with Area under Precision-Recall Curve</h3>
<p>With the Area under Precision-Recall Curve (AUPR), it is possible to evaluate precision and recall using only one value. Using the R script <a href="scripts/../../scripts/visPRCurve.R">visPRCurve.R</a>, the values in <a href="../attachments/tables/table6.md" title="Table 6: AUPR values, Default Database"><em><strong>Table 6</strong></em></a> and <a href="../attachments/tables/table7.md" title="Table 7: AUPR values, Custom Database"><em><strong>Table 7</strong></em></a> are calculated. <br>
For this metric, higher values are better than lower values, ranging from 0-1.0. Since Centrifuge assigns a read to up to five species, the abundances are calculated slightly different: the number of reads assigned to one entry, i.e. species or another taxonomic rank, is divided by the total number of assignments which can be greater than the number of reads. This is important for the threshold that is considered for precision and recall. <br></p>
<p>Although Diamond and Kaiju both are protein-based tools, Diamond seems to achieve a better AUPR for GridION364 with a score of 1.0, which equals a perfect classification (<a href="../attachments/tables/table6.md" title="Table 6: AUPR values, Default Database"><em><strong>Table 6</strong></em></a>). This means that for all thresholds between 0-1, Diamond classified all species correctly. i.e. the precision is always 1 for all recall values until a very high threshold. where the recall is one then. Kaiju achieves a score of 0.73 (for both CS Even samples) and performs slightly worse with 0.39 for GridION366, for which Diamond scores 0.51. Kaiju’s scores for the PromethION samples are in the same scope as those for its GridION samples. However, Kaiju scores lowest for the CS Log samples. Other tools achieve at least a score of 0.51, the majority of tools even exceed 0.65. An explanation for this behaviour might be the different types of databases since Diamond and Kaiju perform based on protein sequences. Both tools allow for the elongation of exact matches, therefore should probably perform better for long reads than tools with static and greater k-mer lengths <a href="https://doi.org/10.1186/s12859-021-04089-5" title="Fan, J., Huang, S., &amp; Chorlton, S. D. (2021). BugSeq: a highly accurate cloud platform for long-read metagenomic analyses. *BMC bioinformatics*, 22(1), 1-12.">[25]</a>. Greater k-mer lengths achieve higher precision, but lower recall. For higher recall, smaller k-mer matches should be used <a href="https://doi.org/10.1101%2Fgr.210641.116" title="Kim, D., Song, L., Breitwieser, F. P., &amp; Salzberg, S. L. (2016). Centrifuge: rapid and sensitive classification of metagenomic sequences. *Genome research*, 26(12), 1721-1729.">[7]</a>. Nevertheless, the differences for GridION366 might be caused by different seed lengths. Kaiju has a default minimum length of 11 amino acids, whereas Diamond uses blastx for the computation of alignments and blastx has a default word length of 6 positions (This value is used with the blastx <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=blastx&amp;PAGE_TYPE=BlastSearch&amp;LINK_LOC=blasthome" title="BLASTx Home">interface</a>, no other source could be found that states the default length Diamond uses). Since a longer seed sequence results in less sensitivity, Diamond is more sensitive than Kaiju which could explain the differences in the AUPR score. <br>
That Diamond achieves the results of a perfect classifier with GridION364 seems counterintuitive due to the later discussed abundances of the species. In contrast to Centrifuge and Kraken2, Diamond is not able to assign more than 1% of reads to the majority of present species in the sample (<a href="../attachments/supplements/table_s1.md" title="Supplements, Table 1"><em><strong>Table-S1</strong></em></a>, supplements). However, this outcome is computed with the PPROC packages of R as well as validated with the sklearn.metrics package of Python (<a href="../scripts/calcPRCurve.py">script</a>). It therefore strengthens the use of more than one metric to evaluate the performance of classifiers (<a href="../attachments/supplements/figure_s1.md" title="Supplements, Figure 1"><em><strong>Figure-S1</strong></em></a>, supplements). <br>
Except for CCMetagen, the tools perform better with the CS Even samples, Centrifuge, CLARK and Kraken2 even achieve an AUPR score of 1.0, which equals the results of a perfect classification. That those results are not achieved for the CS Log samples appear to be reasonable since certain species have extremely low abundances and hence only produce a limited number of reads in the sample. <br>
CCMetagen has a score of 0.69 for GridION364 and 0.78 for GridION366, the results for CLARK for the CS Log samples range from 0.69 to 0.73, the CS Even samples achieve the same AUPR of 1.0. The best score for the PromethION367 sample is achieved with CLARK, it outperforms the AUPRs of the other tools concerning these samples - it is the highest AUPR for the CS Log samples with 0.73. With a default k-mer length of 31 nt it is expected to be more sensitive than tools with longer k-mer lengths (Kraken2: 35 nt), this might be a reason why the AUPR on the CS Log samples is higher, whereas the AUPRs for the CS Even samples are the same. <br>
Considering Kraken2 and Centrifuge and the CS Log samples, Kraken2 achieves better results with values of 0.672 and 0.696 for GridION and PromethION, whereas Centrifuge achieves scores of 0.606 and 0.614, respectively. This is surprising since Centrifuge has a minimum match length of 22 nt, but elongates the matches, whereas Kraken2 uses a fixed k-mer length of 35 nt. Therefore, Centrifuge’s sensitivity should be greater than Kraken2’s. <br></p>
<p>However, there does not seem to be a trend whether one of the approaches works best with the long reads of this dataset, the sequencing depth does not seem to influence the AUPR on a big scale either. Most tools achieve slightly better results with the PromethION samples, but the improvements are small (Kaiju: 0.3873 to 0.3897, Centrifuge: 0.606 to 0.614, CLARK: 0.69 to 0.73, Kraken2: 0.672 to 0.697). <br>
A trend that can be observed is that Diamond and Kaiju, both using protein classification, seem to perform better for CS Even samples with their default database. <br>
The missing fungal sequences in the databases do not interfere with the precision-recall curves (e.g. <a href="../attachments/supplements/figure_s2.md" title="Supplements, Figure 2"><em><strong>Figure-S2a, Figure-S2b</strong></em></a>). <br></p>
<p>BugSeq is only briefly compared to the other tools since it is a cloud service. The results are based on the default database used by BugSeq. The tool achieves an AUPR of 1.0 for the CS Even sample and and a slightly lower value of 0.959 for the CS Log samples (<a href="../attachments/tables/table6.md" title="Table 6: AUPR values, Default Database"><em><strong>Table 6</strong></em></a>). This means the results are better for the GridION366 sample than the scores the other classifiers achieved. This might be explained by the structure of this cloud service: BugSeq is a pipeline with several tools which are refined and mixed to achieve the best results, even without custom changes of parameters.</p>
<table>
<thead>
<tr>
<th></th>
<th>Diamond</th>
<th>Kaiju</th>
<th>CCMetagen</th>
<th>Centrifuge</th>
<th>CLARK</th>
<th>Kraken2</th>
<th>BugSeq</th>
</tr>
</thead>
<tbody>
<tr>
<td>GridION364</td>
<td>1.0</td>
<td>0.7253469</td>
<td>0.6887791</td>
<td>1.0</td>
<td>1.0</td>
<td>1.0</td>
<td>1.0</td>
</tr>
<tr>
<td>PromethION365</td>
<td>-</td>
<td>0.7253469</td>
<td>-</td>
<td>1.0</td>
<td>1.0</td>
<td>1.0</td>
<td>-</td>
</tr>
<tr>
<td>GridION366</td>
<td>0.5103104</td>
<td>0.3873827</td>
<td>0.7760189</td>
<td>0.6062113</td>
<td>0.6914607</td>
<td>0.6724297</td>
<td>0.9588289</td>
</tr>
<tr>
<td>PromethION367</td>
<td>-</td>
<td>0.3896924</td>
<td>-</td>
<td>0.6138137</td>
<td>0.7315841</td>
<td>0.6965986</td>
<td>-</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table6.md" title="Table 6: AUPR values, Default Database"><em><strong>Table 6: AUPR values, Default Database.</strong></em></a> This table shows the calculated AUPR for the different tools and samples using their default database. Note that CCMetagen and Diamond are not able to perform on the PromethION samples, therefore those values are missing. There is no trend to be observed regarding the AUPR and different sequencing depths. However, some tools seem to perform better with the CS Even samples (e.g. Diamond, Centrifuge, Kraken2), whereas Kaiju, CCMetagen and CLARK seem to perform better with the CS Log samples. The corresponding plots can be seen in <a href="../attachments/supplements/figure_s9.md" title="Supplements, Figure 9"><em><strong>Figure-S9</strong></em></a> or <a href="../stats/pics/prc" title="Folder of Precision-Recall-Curves">here</a>. <br> <br></p>
<table>
<thead>
<tr>
<th></th>
<th>Diamond</th>
<th>Kaiju</th>
<th>CCMetagen</th>
<th>Centrifuge</th>
<th>CLARK</th>
<th>Kraken2</th>
</tr>
</thead>
<tbody>
<tr>
<td>GridION364</td>
<td>0.902987</td>
<td>0.7602105</td>
<td>-</td>
<td>0.69504</td>
<td>0.5257158</td>
<td>0.8988399</td>
</tr>
<tr>
<td>PromethION365</td>
<td>-</td>
<td>0.7602105</td>
<td>-</td>
<td>0.6879514</td>
<td>0.523079</td>
<td>0.8700717</td>
</tr>
<tr>
<td>GridION366</td>
<td>-</td>
<td>0.3398543</td>
<td>-</td>
<td>0.361886</td>
<td>0.5102082</td>
<td>0.5979277</td>
</tr>
<tr>
<td>PromethION367</td>
<td>-</td>
<td>0.4339642</td>
<td>-</td>
<td>0.3265234</td>
<td>0.5100713</td>
<td>0.6200843</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table7.md" title="Table 7: AUPR values, Custom Database"><em><strong>Table 7: AUPR values, Custom Database.</strong></em></a> This table shows the calculated AUPR for the different tools and samples using their custom database. Note that Diamond is not able to perform on the PromethION samples, therefore those values are missing. The scores for Kraken2 and Centrifuge are lower than those achieved with the default database. However, for Kraken2, the CS log samples have higher values, whereas Centrifuge scrores higher with the CS Even samples. Kaiju improves for the CS Log samples, but results in lower scores for CS Even samples. CLARK is able to achieve a AUPR between 0.51 and 0.53 over all for samples, which is the poorest result. The corresponding plots can be seen in <a href="../attachments/supplements/figure_s10.md" title="Supplements, Figure 10"><em><strong>Figure-S10</strong></em></a>) or <a href="../stats/pics/prc" title="Folder of Precision-Recall-Curves">here</a>. There are no values for CCMetagen since KMA is not able to build an index due to memory limits and it is not possible to evaluate results for GridION366 and Diamond because of the same reason. <br> <br></p>
<p>The results of AUPR analysis regarding the custom databases shows a different outcome (<a href="../attachments/tables/table7.md" title="Table 7: AUPR values, Custom Database"><em><strong>Table 7</strong></em></a>). Centrifuge and Kraken2 are no longer able to perform like a perfect classifier. The AUPRs for Centrifuge range from 0.33 to 0.7, which is a deterioration compared to the previously discussed results. The degradation for Kraken2 is not that strong, the values for the CS Log samples are similar to the scores Kraken2 achieved on the CS Log samples with the default database, ranging from 0.598 to 0.62. The CS Even samples reach AUPRs of 0.899 and 0.87 for GridION and PromethION, respectively. An explanation for this worsening might be the custom database since it is the only change between runs, although both default databases are based on RefSeq sequences as are the custom databases. Variations might be due to different building timepoints and therefore distinctions in the included sequences. The index for Centrifuge is from 2018, whereas the Kraken2 database is generated in 2019, but the custom database relies on sequences that are last updated in 2021. It is possible that for some species, more isolates or sequences, in general, are annotated by now which might increase the number of possible hits and therefore might decrease the certainty of some tools. These changes also lead to a loss in balanced accuracy. With the default database, Kraken2 has an accuracy of 1.0 for GridION364, but with the custom database, Kraken2 only reaches 0.83, this is the same for PromethION365. A similar difference can be seen for Centrifuge. Balanced accuracy is the proportion of added False Positive Rate and True Negative Rate divided by 2. It is used when the classes for classification are imbalanced, i.e. there are more species not in the sample than species that are in the sample. <br>
For Kaiju, the CS Even samples achieve an AUPR of 0.76 both, which is an improvement for both samples. This is different for the CS Log samples: The GridION sample deteriorates, whereas the PromethION sample improves. This slight improvement might be due to the greater sequencing depths of the PromethION samples. Overall, the CS Even samples approximate the results of Kraken2 and Centrifuge, the differences are not as great as for the default database. Diamond, however, seems to perform best for GridION364 regarding the AUPR and custom database. The score of 0.903 is slightly higher than the corresponding value Kraken2 achieves (0.899). However, as discussed earlier, this result seems counterintuitive due to the achieved abundances by Diamond. The underlying sequences for the protein databases are the same for the bacterial sequences, therefore differences in the included sequences are only possible for the fungal sequences. The smaller amount of changes within the fungal sequences may account for the slight degradation, even leading to an improvement for Kaiju and the CS Even samples, because the differences between default and custom database are not as extreme as for the other tools. <br>
Due to memory limits, the index for CLARK is built with CLARK-l which is a memory-sensitive version of CLARK. The corresponding k-mers have a length of 27 nt instead of 31 nt, the database is described as sparse, hence the memory sensitivity. The results should be understood as a draft or approximation of the results CLARK would achieve. It is not possible to run CLARK on the sparsed database. Although the results are comparingly bad, they have to be evaluated considering the use of CLARK-l. The deterioration between default and custom database is particularly noticeable with the CS Even samples. Previously, the tool achieves an AUPR of 1.0. Here, the AUPR is 0.526 and 0.523 for GridION and PromethION, respectively. The values for the CS Log samples are in a similar scope (0.51 each) and therefore, do not show a similar deterioration. Those values are higher than the corresponding values of Kaiju and Centrifuge. The worsening might be explained by the sparsed database, but poorer results for the CS Log samples would have been expected. It can be seen in the later discussed classification outputs as well. The majority of bacterial species of the samples can be analysed, but several species are not present in the underlying dataset.<br></p>
<p>Furthermore, the classification of GridION364 with Diamond and the custom database results in an AUPR of 0.903, which is the best-achieved value. However, as explained before, this seems counterintuitive due to the later discussed abundances. <br>
It is not possible to evaluate the AUPR of GridION366 because of disk and memory limitations.</p>
<p>Due to the beforementioned limits in memory, it is not abe to run CCMetagen with the custom database. This might be due to the increased size of the database. <br></p>
<p>It seems like the tools perform better on their (downloaded) default databases/indices. However, Kraken2 and Kaiju show the least deterioration between the two databases although Kraken2 performs best overall samples and databases, regarding the AUPR. This might be explained due to Kaiju not being specially designed for long reads, whereas Kraken2 is supposed to be able to work with those reads as well. In conclusion, it seems like Kraken2 and Kaiju achieve the best scores regarding the AUPR with Centrifuge following. Diamond does not seem like a good option due to the long runtimes for samples with greater sequencing depth, the AUPRs, however, seem good. <br></p>
<!-- Most of the tools are also used in a benchmarking study for short reads by Simon H. Ye et al. [[10]](https://doi.org/10.1016/j.cell.2019.07.010 "Simon, H. Y., Siddle, K. J., Park, D. J., & Sabeti, P. C. (2019). Benchmarking metagenomics tools for taxonomic classification. *Cell*, 178(4), 779-794."). In that study, most tools achieved scores between 0.9 and 1.0 with their default database, except for Centrifuge which scored between 0.3 and 0.4. Using a RefSeq CG database, all tools are able to score between 0.9 and 1.0. This stands in contrast to the results presented here. The custom databases seem to result in inferior AUPR scores than the classifications with the default databases. <br> -->
<p>It has to be considered that only one dataset and four samples are not enough to fully access the performance of the classification tools.</p>
<h3 id="benchmarking-with-abundance-profile-similarity">Benchmarking with Abundance Profile Similarity</h3>
<p>In this section, the abundance profile similarity is discussed as well as the existence of differences for the different types of species, i.e. if some tools deal better with Gram+ or Gram- samples.</p>
<table>
<thead>
<tr>
<th></th>
<th>Diamond</th>
<th>Kaiju</th>
<th>CCMetagen</th>
<th>Centrifuge</th>
<th>CLARK</th>
<th>Kraken2</th>
<th>BugSeq</th>
</tr>
</thead>
<tbody>
<tr>
<td>GridION364</td>
<td>0.3227856</td>
<td>0.2524564</td>
<td>0.2612868</td>
<td>0.0918381</td>
<td>0.1563444</td>
<td>0.0993529</td>
<td>0.8995511</td>
</tr>
<tr>
<td>PromethION365</td>
<td>-</td>
<td>0.2545771</td>
<td>-</td>
<td>0.0958232</td>
<td>0.1560923</td>
<td>0.1022717</td>
<td>-</td>
</tr>
<tr>
<td>GridION366</td>
<td>0.7078238</td>
<td>0.3219797</td>
<td>0.1864602</td>
<td>0.0698712</td>
<td>0.0806941</td>
<td>0.0923067</td>
<td>0.0412768</td>
</tr>
<tr>
<td>PromethION367</td>
<td>-</td>
<td>0.3207758</td>
<td>-</td>
<td>0.0903556</td>
<td>0.0978355</td>
<td>0.1056878</td>
<td>-</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table8.md" title="Table 8: Abundance Profile Similarities, Default Database"><em><strong>Table 8: Abundance Profile Similarities, Default Database.</strong></em></a> This table shows the Abundance Profile Similarity scores for the different tools and samples, rounded to seven digits. Since this value is based on the L2 distance, lower values are better than higher ones. Hence, both protein samples show poorer results for the CS Log samples, whereas the other tools show better values for the CS Log samples. Note that the APS for CCMetagen and Kraken2 include the fungal abundances, whereas the APS for the other tools do not take those into account here. <br> <br></p>
<p>Regarding the Abundance Profile Similarity (APS), lower values are better than high values, since the metric is based on the L2 distance, the values can range between 0-1.0. The APS scores for the default databases are generated with or without the fungal abundances depending on those species present in the default database. For the custom database, all abundances are considered. <br></p>
<p>This means that, regarding the APS, Kraken2 and Centrifuge performed best with their default databases, i.e. classified the species of the CS samples with similar proportions. Centrifuge scored the best result with GridION366 and an APS value of 0.0699 (<a href="../attachments/tables/table8.md" title="Table 8: Abundance Profile Similarities, Default Database"><em><strong>Table 8</strong></em></a>). In general, Centrifuge’s outcome seems to be closest to the ground truth abundances, the average APS of all four samples is the lowest with 0.087, whereas Kraken2 has an average score of 0.0999. This might be due to the classification approach.
Especially the PromethION samples result in less abundance profile similarity, although the difference is small, e.g. Kraken2 has a score of 0.0994 for GridION364 and 0.1023 for PromethION365.  <br>
CLARK’s APS scores show an improvement for CS Log samples, such a difference could not be observed for Centrifuge or Kraken2. The CS Even samples for CLARK achieve an average APS of 0.156, whereas the average score is 0.0893 with the CS Log samples. This contrast might be explained by the logarithmic distribution of the present species. Since there are only a few species with high abundances and therefore with a predominant number of reads, the classification of reads with a sufficient abundance is more likely for those few species than to correctly reconstruct the abundances of all twelve species for the CS Even samples. Differences in the lower abundance values might not have that big of an impact. <br>
The protein-based tools Kaiju and Diamond seem to do worst with the correct abundances of species, although CCMetagen scores high values as well. In contrast to CLARK, Diamond shows a great degradation between the CS Even GridION sample and the CS Log GridION sample. The APS score for GridION364 is 0.323, which ranges in another scope than CLARK, Centrifuge and Kraken2. However, it is similar to the values Kaiju and CCMetagen compute. Kaiju’s APS scores range from 0.2525 to 0.322 with the higher values for CS Log samples. CCMetagen performs slightly better with 0.2613 and 0.1865 for GridION364 and GridION366, respectively. The GridION366 score of Diamond scores the highest, regarding the default databases, with 0.7078. This strengthens the possibility that an alignment-based approach is not the best way for taxonomic classification. Kaiju uses an FM-Index and performs slightly better, this would support the assumption why Centrifuge achieves better scores than Kraken2. Therefore, the worsening of APS scores might not be due to the protein classification, but due to the alignment approach. However, CCMetagen, which uses an alignment approach as well due to KMA, scores better than Diamond. Hence, the combination of alignment and protein might be a cause for the bad APS. <br></p>
<p>Considering BugSeq, the APS for both GridION samples are in a similar scope as Centrifuge and CLARK, the APS of GridION366 especially is the best score with 0.0413 for the default database. The APS of GridION364 is only slightly lower than the corresponding scores for Centrifuge and Kraken2 with BugSeq having a score of 0.08996 (<a href="../attachments/tables/table8.md" title="Table 8: Abundance Profile Similarities, Default Database"><em><strong>Table 8</strong></em></a>).</p>
<table>
<thead>
<tr>
<th></th>
<th>Diamond</th>
<th>Kaiju</th>
<th>CCMetagen</th>
<th>Centrifuge</th>
<th>CLARK</th>
<th>Kraken2</th>
</tr>
</thead>
<tbody>
<tr>
<td>GridION364</td>
<td>0.3236971</td>
<td>0.3007596</td>
<td>-</td>
<td>0.2272981</td>
<td>0.2970810</td>
<td>0.1769953</td>
</tr>
<tr>
<td>PromethION365</td>
<td>-</td>
<td>0.3017296</td>
<td>-</td>
<td>0.229281</td>
<td>0.2925519</td>
<td>0.1779121</td>
</tr>
<tr>
<td>GridION366</td>
<td>-</td>
<td>0.476431</td>
<td>-</td>
<td>0.1671362</td>
<td>0.6879505</td>
<td>0.0870177</td>
</tr>
<tr>
<td>PromethION367</td>
<td>-</td>
<td>0.4707885</td>
<td>-</td>
<td>0.1931905</td>
<td>0.6799135</td>
<td>0.101036</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table9.md" title="Table 9: Abundance Profile Similarities, Custom Database"><em><strong>Table 9: Abundance Profile Similarities, Custom Database.</strong></em></a> This table shows the Abundance Profile Similarity scores for the different tools and samples, rounded to seven digits. Since this value is based on the L2 distance, lower values are better than higher ones. Hence, both protein samples show poorer results for the CS Log samples, whereas the other tools show better values for the CS Log samples. Although this trend stays the same for default and custom database, there is a strong degradation, all tools show values above 0.1, with the exception for Kraken2's GridION366 sample. <br> <br></p>
<p>The APS for the custom database show a different outcome (<a href="../attachments/tables/table9.md" title="Table 9: Abundance Profile Similarities, Custom Database"><em><strong>Table 9</strong></em></a>). This means that if the tools have performed well with very low values below 0.1 with the default database, only Kraken2 can achieve such a result for one sample using the custom database. All other tools show significantly higher values. Overall, Kraken2 achieves the lowest values ranging from 0.087 to 0.178, which is on average about 0.05 points better than the second-best values which are achieved by Centrifuge. Those values range from 0.167 to 0.227. The worsening of the results might be explained by the custom database which includes more species due to the added fungal sequences. Since Kraken2's default database already included fungal sequences, this might explain why the tool achieves comparable good results. This might contradict the assumption that FM-index-based approaches can achieve better results than k-mer-based approaches.<br>
However, the protein-based tools Diamond and Kaiju show the least worsening, especially Diamond and the GridION364 sample achieve similar values with 0.32279 using the default database and 0.3237 with the custom database. As mentioned before, no evaluation of Diamond's classification performance for GridION366 is possible. Kaiju's APS deteriorate more, the CS Even samples now achieve values of 0.3 (Default: 0.25) and the CS Log samples have an APS of 0.476 and 0.471, respectively. However, the values range in a similar scope as those of CLARK, they are even slightly better for the CS Log samples. This might support the beforementioned theory that the protein-based methods might not be the reason for the poorer results compared to the tools based on DNA sequences. There does not seem to be an explanation based on the classification approaches since Kaiju and Centrifuge both use an FM-Index but achieve varying results concerning performance compared to the other tools. Additionally, Kraken2 and CLARK both use a k-mer-based approach and also achieve results in a different scope. This might be due to different k-mer lengths (Kraken2: 31 nt, CLARK-l: 27 nt) and it has to be considered that CLARK is run in a memory-efficient way with a sparsed database, which might be an explanation for the poor results, especially considering the CS Log samples. <br>
Another aspect to consider is that the metrics in this project are based on species-level classification, but some tools identify up to the strain or subspecies level (e.g. Diamond, Kraken2) which might cause less APS in general, although considering corresponding strains or subspecies could lead to the expected amount of reads for each species. An example is <em>Bacillus subtilis</em> and <em>Bacillus subtilis subsp. spizizenii</em> in the Kraken2 or CCMetagen results. <br></p>
<p>To conclude the analysis of the APS, it can be said that Kraken2 and Centrifuge seem to perform best with the default database as well as the custom database, although there is a deterioration. CLARK might be an option as well if the required memory is provided to run the tool with the full index and not a sparse one. However, based on these results for the two databases and the one dataset, it seems like Kraken2 and Centrifuge are the tools to use if only terminal tools are considered. BugSeq seems an option if the files do not exceed 10 Gb in size. For greater files, the tool is not free anymore.</p>
<!--Drinlassen?? 
Considering the paper of Simon H. Ye et al. [[10]](https://doi.org/10.1016/j.cell.2019.07.010 "Simon, H. Y., Siddle, K. J., Park, D. J., & Sabeti, P. C. (2019). Benchmarking metagenomics tools for taxonomic classification. *Cell*, 178(4), 779-794."), most tools achieve scores in the range of 0-0.1 using their default database, except for Centrifuge. Considering the used RefSeq CG database, there is only a noticeable change for Centrifuge which achieves scores between 0-0.1, too. This stands again in contrast to the results presented here. Although the values using the default database are considerable good and similar to that study, for Centrifuge even better, the use of the custom database leads to deterioration for most tools which can not be observed by Simon H. Ye et al. -->
<h4 id="differences-for-gram-positive-and-gram-negative-species">Differences for Gram-positive and Gram-negative species</h4>
<p>In hindsight of the different properties of species, this section deals with possible differences of the classifiers regarding the gram staining of bacteria. The gram stain of the bacterial species in the samples can be seen in <a href="../attachments/tables/table1.md" title="Table 1: General Information about the Used Species"><em><strong>Table 1</strong></em></a>. Only the CS Even samples are discussed since the proportional abundances allow for comparisons regarding the preferences of classification tools, whereas the logarithmic abundances do not allow for that. <br></p>
<p>The abundances for the species in the samples classified using the default databases can be seen in <a href="../attachments/supplements/table_s1.md" title="Supplements, Table-S1"><em><strong>Tables-S1</strong></em></a>, <a href="../attachments/supplements/table_s2.md" title="Supplements, Table-S2"><em><strong>-S2</strong></em></a>, <a href="../attachments/supplements/table_s3.md" title="Supplements, Table-S3"><em><strong>-S3</strong></em></a> and <a href="../attachments/supplements/table_s4.md" title="Supplements, Table-S4"><em><strong>-S4</strong></em></a>. The protein-based tools Kaiju and Diamond associate a minority of reads to the species in the sample. For Diamond (GridION364), the percentage of assigned reads is between 0 and 1% except for <em>Listeria monocytogenes</em> with 2.89%. However, considering the genus level, Diamond is able to identify more reads belonging to a genus included in the dataset. The genera with the most abundances are <em>Listeria</em>, (9.459%), <em>Staphylococcus</em> (7.004%) and <em>Pseudomonas</em> (4.54%), see <a href="../attachments/supplements/figure_s3.md" title="Supplements, Figure-S3"><em><strong>Figure-S3</strong></em></a>. Two of the corresponding species in this dataset are gram-positive, whereas <em>Pseudomonas aeruginosa</em> is gram-negative, this proportion is applicable for the low abundances genera as well, hence there does not seem to be a trend which gram-species can be better classified with Diamond. <br>
Kaiju on the other hand assigns more reads to gram-positive species (4.151% - 12.503% of abundance), whereas gram-negative species achieve fewer assigned reads (1.104% - 2.124%) in GridION364. This is also true for the PromethION sample, the classified gram-positive species abundances range from 3.98% to 12.033%, whereas the gram-negative species have 1.111% to 2.101% of reads assigned to (<a href="../attachments/supplements/table_s2.md" title="Supplements, Table-S2"><em><strong>Table-S2</strong></em></a>, <a href="../stats/pics/gridion364_default.kaiju.piechart.jpeg" title="(a) GridION364: Piechart for Classification Results of Kaiju (default)"><em><strong>Figure 3a</strong></em></a>). <br>
The four dominant species in the classification result of CCMetagen are <em>Bacillus spizizenii</em>, <em>Listeria monocytogenes</em>, <em>Limosilactobacillus fermentum</em> and <em>Staphylococcus fermentum</em>, which are all gram-positive species, two of the three remaining species with abundances ranging from 1.795% to 3.387% are gram-negative. This might imply a preference for gram-positive species. <br>
The classification results for the CS Even samples seem to be balanced for Centrifuge, there are no great differences in abundances. However, the three species with around 5% abundance (<em>Escherichia coli</em>, <em>Pseudomonas aeruginosa</em> and <em>Salmonella enterica</em>) are all gram-negative species. But since Kraken2 has a similar distribution of reads, this might not necessarily be due to the gram stain. Additionally, all three species have less abundance in the ground truth than expected (around 9%, <a href="../attachments/tables/table1.md" title="Table 1: General Information about the Used Species"><em><strong>Table 1</strong></em></a>). <br>
The abundances after classification with CLARK are similar to Kraken2 and Centrifuge. The three species with the lowest abundances are those three gram-negative species. This means, either the tools have a preference for gram-positive species or the abundance of the gram-negative species is generally lower by chance than the abundances of the other species in the sample. The first assumption is supported by <em>Lactobacillus fermentum</em> (gram-positive): The estimated abundance in the ground truth is also lower than expected, but the tools associate a higher amount of reads with it. This stands in contrast to the theory that the estimated lower abundance of reads (compared to the expectation) in the ground truth is the reason for the lower percentage of assignment. <br> <br></p>
<p>The different abundances achieved with the custom database for the species of the sample can be seen in the supplements (<a href="../attachments/supplements/table_s10.md" title="Supplements, Table-S10"><em><strong>Table-S10</strong></em></a>, <a href="../attachments/supplements/table_s11.md" title="Supplements, Table-S11"><em><strong>-S11</strong></em></a>, <a href="../attachments/supplements/table_s12.md" title="Supplements, Table-S12"><em><strong>-S12</strong></em></a>, <a href="../attachments/supplements/table_s10.md" title="Supplements, Table-S10"><em><strong>-S13</strong></em></a>). Considering that run of classification and GridION364, there are no noticeable changes for Diamond. This is true for Kraken2 and Centrifuge as well, except <em>Bacillus subtilis</em> which only has an abundance of 4.912% with the custom database instead of 17.51% for Kraken2 using the default database. The values for Centrifuge are similar. <em>Bacillus subtilis</em> is a gram-positive species, the fallen abundance probably is not due to that or other gram-positive species should have a loss of abundance as well. This is not the case, <em>Escherichia coli</em> and <em>Staphylococcus aureus</em> even have a gain in the abundance of reads associated with the species. CLARK has a general deterioration regarding the species in the sample (possibly due to the sparse database), there does not seem to be a connection. However, CLARK classified several other species that are not in the sample, but the proportion of gram-positive and -negative species is similar. <br>
Kaiju's abundances are mixed. For some species, the values differ and are overall slightly poorer. The greatest changes can be observed for <em>Bacillus subtilis</em> (from 0.81% to 0.21% for default and custom database, respectively), <em>Enterococcus faecalis</em> (10.27% and 2.37%) and <em>Lactobacillus fermentum</em> (12.5% and 5.54%), which are all gram-positive species. However, there are three gram-negative species with great changes as well (<em>Salmonella enterica</em>, <em>Escherichia coli</em> and <em>Pseudomonas aeruginosa</em>), but those changes are not as severe. This could hint at a preference for gram-negative species which contradicts the already discussed preference for gram-positive species with the default database. It is more likely that the lower abundances are due to the custom database. <br>
This is true for PromethION365 too. <br></p>
<p>BugSeq is not considered here further since the classification results, regarding the abundance, correspond to the estimated and therefore expected abundances for all species, hence a preference can not be detected. <br></p>
<p>It seems like that especially the protein-based tools have a preference for gram-positive species, although it might be possible that this trend is only due to the actual abundances within the sample. Centrifuge, Kraken2 and CLARK seem to have a slight preference for gram-negative species, at least while using the default database. This can be observed for the custom database as well, but only with Centrifuge and Kraken2. It might not be observed for CLARK because of the sparse database.</p>
<h3 id="computational-requirements-runtime-and-database-creation">Computational Requirements: Runtime and Database Creation</h3>
<table>
<thead>
<tr>
<th></th>
<th>Diamond</th>
<th>Kaiju</th>
<th>KMA (CCMetagen)</th>
<th>Centrifuge</th>
<th>CLARK</th>
<th>Kraken2</th>
</tr>
</thead>
<tbody>
<tr>
<td>GridION364</td>
<td>2 days, 15:16:56.140675</td>
<td>3:30:55.818875</td>
<td>0:58:33.332763 (0:00:19.790294)</td>
<td>3:28:34.545058</td>
<td>0:42:52.407070</td>
<td>0:08:55.109309</td>
</tr>
<tr>
<td>PromethION365</td>
<td>-</td>
<td>1 day, 9:14:36.672953</td>
<td>-</td>
<td>1 day, 14:05:20.470530</td>
<td>5:35:37.829886</td>
<td>1:26:06.232660</td>
</tr>
<tr>
<td>GridION366</td>
<td>4 days, 10:02:19.511836</td>
<td>3:40:21.257391</td>
<td>2:03:00.949100 (0:00:47.213880)</td>
<td>4:45:25.614299</td>
<td>0:38:11.365624</td>
<td>0:11:17.788488</td>
</tr>
<tr>
<td>PromethION367</td>
<td>-</td>
<td>1 day, 4:03:14.998929</td>
<td>-</td>
<td>1 day, 9:46:48.769590</td>
<td>5:24:34.567883</td>
<td>1:27:05.043035</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table10.md" title="Table 10: Overview of Time Consumption for the Default Runs"><em><strong>Table 10: Overview of Time Consumption for the Default Runs.</strong></em></a> Time is given as hh:mm:ss and if needed, the number of days is stated explicitly in front. Cells with a dash symbolize runs that did not start or took unreasonable much time to start. The time benchmarks can be found <a href="../attachments/paths.md" title="Benchmark Files, Default">here, Line 26</a>. <br> <br></p>
<p>Runtime is a secondary metric to evaluate tools. It has no explicit effect on the output, but tools that classify in efficient time have an advantage considering the increasing datasets, and sometimes, parameters regarding the performance (especially sensitivity) increase the runtime. <br>
Looking at all the classifiers, Kraken2 is the fastest with the default database (<a href="../attachments/tables/table10.md" title="Table 10: Overview of Time Consumption for the Default Runs"><em><strong>Table 10</strong></em></a>). The GridION samples took a few minutes (0:08:55, 0:11:17), the PromethION samples about one and a half hours (1:26:06 and 1:27:05 for PromethION365 and PromethION367, respectively). Thus, even the PromethION samples are classified faster with Kraken2 than the GridION samples with most other tools. <br>
Diamond needs about the longest for the GridION samples, the calculations for the PromethION samples stopped after about 14 days without a result and were not restarted because of this high computing time. KMA/CCMetagen also has problems ([Error: 28](../attachments/kma.promethion.error&quot;Log, KMA and promethion367 (Line 25)&quot;)) with the PromethION samples, which contain on average ten times as many reads as the GridION samples. <br>
Centrifuge and Kaiju need up to 4:45:25 hours (Centrifuge, GridION366) for the GridION samples and up to 1 day, 14:05:20 hours (Centrifuge, PromethION365) for the PromethION samples with Kaiju being a little faster (GridION366: 3:40:21, PromethION365: 1 day, 9:14:36). <br>
CLARK's performance is between those extremes. It needs 40 minutes for the GridION samples (0:42:52 for GridION364 and 0:38:11 for GridION366, respectively) and about five and a half hours for the PromethION samples. <br></p>
<p>This trend can be observed within the custom databases as well (<a href="../attachments/tables/table11.md" title="Table 11: Overview of Time Consumption for the Custom Runs"><em><strong>Table 11</strong></em></a>). Kraken2 is still the fastest with similar times. The runtimes for Kaiju do not change much either with the custom database. This is true for Diamond and GridION364 as well. <br>
Centrifuge is able to classify the GridION samples in around five hours (e.g. GridION364: 5:09:32.420430), whereas the PromethION samples now need up to two days (PromethION365: 2 days, 6:49:22.551248, PromethION367: 2 days, 3:59:35.103469). Another deterioration can be seen with CLARK for the PromethION samples. Those samples took about 5.5 h with the default database, but up to 15 h with the custom database. The runtime for GridION samples is similar for both databases.
The slower performance might be explained by the greater number of species in the custom database. The Centrifuge and CLARK default databases do not contain fungal sequences. However, other tools do not show such degradation in runtime.
<br></p>
<table>
<thead>
<tr>
<th></th>
<th>Diamond</th>
<th>Kaiju</th>
<th>KMA (CCMetagen)</th>
<th>Centrifuge</th>
<th>CLARK</th>
<th>Kraken2</th>
</tr>
</thead>
<tbody>
<tr>
<td>GridION364</td>
<td>2 days, 17:38:23.548931</td>
<td>3:05:24.622132</td>
<td>-</td>
<td>5:09:32.420430</td>
<td>0:48:06.177309</td>
<td>0:12:03.455402</td>
</tr>
<tr>
<td>PromethION365</td>
<td>-</td>
<td>1 day, 7:11:14.124467</td>
<td>-</td>
<td>2 days, 6:49:22.551248</td>
<td>14:17:44.676811</td>
<td>1:26:16.939620</td>
</tr>
<tr>
<td>GridION366</td>
<td>-</td>
<td>3:12:15.641435</td>
<td>-</td>
<td>5:31:10.976138</td>
<td>1:07:13.242530</td>
<td>0:12:31.477754</td>
</tr>
<tr>
<td>PromethION367</td>
<td>-</td>
<td>1 day, 3:12:06.381124</td>
<td>-</td>
<td>2 days, 3:59:35.103469</td>
<td>15:28:16.577327</td>
<td>1:29:48.305022</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table11.md" title="Table 11: Overview of Time Consumption for the Custom Runs"><em><strong>Table 11: Overview of Time Consumption for the Custom Runs.</strong></em></a> Time is given  as hh:mm:ss and if needed, the number of days is stated explicitly in front. Cells with a dash symbolize runs that did not start or took unreasonable much time to start. The time benchmarks can be found <a href="../attachments/paths.md" title="Benchmark Files, Custom">here, Line 27</a>. <br> <br></p>
<p>The last aspect of runtime that needs to be considered is the database creation time (<a href="../attachments/tables/table12.md" title="Table 12: Overview of Time Consumption, Database Creation"><em><strong>Table 12</strong></em></a>). <br>
For CLARK, the building time of the indices is considered, which are built with the first call of the classifier for a new database. Therefore, the time for creating the GridION366 classification outcome includes the time of index creation. Altogether, the time needed is 5:51:51.268086 hours. Subtracting the time for a rerun of classification for that sample, which needed 1:07:13.242530, the index creation needs about 4 hours and 40 minutes. This time can be seen in brackets. It has to be considered that a memory-efficient version of CLARK, CLARK-l, is used for the custom database which results in a sparse database.
Although the classification runtime of Diamond is comparably worse, the time for database creation is fast with about 0:23:52. This is the fastest building time, although the CLARK needs less time for the database, but does need a few hours for index building. Kaiju needs about an hour for the index building. There is no time for Kraken2 since the building of the database took several tries and was finally achieved in several separate steps which did not allow for accurate time measuring.
It was not able to generate a custom database for CCMetagen with KMA due to challenges in memory space. <br></p>
<p>Overall, most tools do not need much time to build their database or indices. However, the fastest performance was shown by Diamond.</p>
<table>
<thead>
<tr>
<th></th>
<th>Diamond</th>
<th>Kaiju</th>
<th>KMA (CCMetagen)</th>
<th>Centrifuge</th>
<th>CLARK</th>
<th>Kraken2</th>
</tr>
</thead>
<tbody>
<tr>
<td>runtime</td>
<td>0:23:52.41719</td>
<td>0:58:52.067620</td>
<td>-</td>
<td>1 day, 14:04:24.423278</td>
<td>0:16:43.803112 (~ 4:40:0)</td>
<td>-</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table12.md" title="Table 12: Overview of Time Consumption, Database Creation"><em><strong>Table 12: Overview of Time Consumption, Database Creation.</strong></em></a> Time is given  as hh:mm:ss and if needed, the number of days is stated explicitly in front. The time benchmarks can be found <a href="../attachments/paths.md" title="Benchmark Files, Database">here, Line 28</a>. <br> <br></p>
<p>The runtime of BugSeq can not be evaluated accordingly since the cloud service offers no information about the time the classification needed and no custom database is build.</p>
<h2 id="classification-validation-with-multi-locus-sequence-typing">Classification Validation with Multi Locus Sequence Typing</h2>
<p>Blasting the housekeeping genes against the samples revealed for each of the chosen genes in every species at least one hit. The results can be seen in the corresponding <a href="../stats/mlst">files</a>. The sequence identities for GridION364 range from 88.831% to 99.045% with an average hit length of 470 nt. The hit lengths vary from 311 nt to 645 nt. The greatest e-value is 4.48e-136, the majority of values is 0.0. <br>
Considering the other CS Even sample, PromethION365, the MLST blast shows sequences identities from 90.13% to 99.52% with an average hit length of 468 nt, which is similar to the hit length for GridION364. The lengths vary from 308 nt to 642 nt. The greatest e-value is similar small with 1.43e-141. <br></p>
<p>Blasting the MLSTs against GridION366 shows that for each gene at least one hit can be achieved. The average sequence identity is 85.98% with values ranging from 65.613% to 99.384%. The average match length is comparatively short with 369 nt (ranging from 27 nt to 760 nt) compared to the CS Even match lengths. The average e-value is 0.31 which again is higher and not as good as the e-values for the CS Even samples. <br>
This can be observed within the PromethION367 samples as well. The sequence identities range from 67.684% to 99.749% with an average of 90.296%, therefore having a wider range of identity. The hit lengths range from 29 nt to 742 nt with an average of 404 nt, which again is shorter than the average hit lengths in the CS Even samples. The average e-value is 0.052. <br></p>
<p>The slightly poorer results for the CS Log samples might be explained by the abundances of species in the sample. <br>
The predominant species in the CS Log samples is <em>Listeria monocytogenes</em>, considering only this species, the average sequence identity for both samples and all genes is 99.143%, which is similar and even better than the result for both CS Even samples (98.618%). Considering a species with nearly no abundance, for example, <em>Escherichia coli</em>, the CS Log samples achieve less sequence identity than the CS Even samples (90.631% and 94.145%, respectively), which supports the hypothesis. <br></p>
<p>However, this method seems to be a valuable step for preprocessing if the species in the metagenomic sample are not known.</p>
<h2 id="classification-results">Classification Results</h2>
<p>The following section shows the species in the diagrams that have an abundance of at least 1%. Reads that were not assigned to a species but other taxa, or are below the 1% mark, are summarized in &quot;others&quot;. The threshold of 1% is chosen although the CS Log samples show lower percentages in abundance for some species.  It is a good trade-off between clarity and visibility of the plots and the number of reads considered for a reasonably certain classification. A larger resolution of the figures can be reached by following the blue figure titles. Due to readability and clarity, no large pictures are included.</p>
<h3 id="diamond">Diamond</h3>
<p>As mentioned before, no results can be obtained with Diamond for the PromethION samples. <br>
Diamond is able to classify 84.23% of reads for the CS Even sample (<a href="../stats/pics/gridion364_default.diamond.piechart.jpeg" title="GridION364: Piechart for Classification Results of Diamond (default)">Figure 1</a>) with the default database (<a href="../attachments/tables/table13.md" title="Table 13: Abundances of classified species, Diamond, Default Database"><em><strong>Table 13</strong></em></a>). However, in the sample sequenced with GridION, Diamond is only able to identify <em>Limosilactobacillus fermentum</em> (previously known as <em>Lactobacillus fermentum</em> <a href="https://doi.org/10.1099/ijsem.0.004107" title="Zheng, J., Wittouck, S., Salvetti, E., Franz, C. M., Harris, H. M., Mattarelli, P., ... &amp; Lebeer, S. (2020). A taxonomic note on the genus Lactobacillus: Description of 23 novel genera, emended description of the genus Lactobacillus Beijerinck 1901, and union of Lactobacillaceae and Leuconostocaceae. *International journal of systematic and evolutionary microbiology*, 70(4), 2782-2858.">[26]</a>) with 4.812% of abundance, <em>Enterococcus faecalis</em> with 1.012% and <em>Listeria monocytogenes</em> with 2.895% on species level. More than 75% of the reads could not be assigned to a species.
Diamond is not able to classify the majority of the species in the sample, with many species having low abundances (<a href="../attachments/tables/table13.md" title="Table 13: Abundances of classified species, Diamond, Default Database"><em><strong>Table 13</strong></em></a>, <a href="../attachments/supplements/table_s1.md" title="Supplements, Table-S1"><em><strong>Tables-S1</strong></em></a>, <a href="../attachments/supplements/table_s3.md" title="Supplements, Table-S3"><em><strong>Table-S3</strong></em></a>).</p>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion364_default.diamond.piechart.jpeg" alt="GridION364: Piechart for Classification Results of Diamond (default)" title="GridION364: Piechart for Classification Results of Diamond (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion364_default.diamond.piechart.jpeg" title="GridION364: Piechart for Classification Results of Diamond (default)">GridION364: Piechart for Classification Results of Diamond (default)</a>. The diagram shows the three species classified with more than 1% of reads assigned: <em>Listeria monocytogenes</em> (2.895%), <em>Enterococcus faecalis</em> (1.012%) and <em>Limosilactobacillus fermentum</em> (4.812%). 75.606% of the reads could not be assigned to a species, but other taxa and 15.676% of the reads could not be assigned at all.</td>
</tr>
<tr>
<td style="text-align:center"></td>
<td style="text-align:left"></td>
</tr>
</tbody>
</table>
<p><em><strong>Figure 1: Classification Results for Diamond, CS Even, Default Database</strong></em> <br> <br></p>
<table>
<thead>
<tr>
<th>Classified Species</th>
<th><em>L. monocytogenes</em></th>
<th><em>E. faecalis</em></th>
<th><em>L. fermentum</em></th>
<th></th>
<th>unclassified</th>
<th>others</th>
</tr>
</thead>
<tbody>
<tr>
<td>CS Even</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION364</td>
<td>2.895</td>
<td>1.102</td>
<td>4.812</td>
<td></td>
<td>15.676</td>
<td>75.606</td>
</tr>
<tr>
<td>CS Log</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION366</td>
<td>18.873</td>
<td>-</td>
<td>-</td>
<td></td>
<td>12.051</td>
<td>69.076</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table13.md" title="Table 13: Abundances of classified species, Diamond, Default Database"><em><strong>Table 13: Abundances of classified species, Diamond, Default Database.</strong></em></a> The table shows the classification results of Diamond for the GridION considering the default database (in %). Diamond is not able to identify all bacterial species in the sample, but is able to identify <em>Listeria monocytogenes</em> as predominant species in the CS Log sample. The amount of reads not assigned to a species or a species with less than 1% or abundance is high with 75.606% and 69.076% for GridION364 and GridION366, respectively. <br> <br></p>
<p>Regarding the CS Log samples and the default database, <em>Listeria monocytogenes</em> is supposed to be the most abundant species in the sample. Diamond assigned 18.873% of reads of the GridION sample to that species <a href="../stats/pics/gridion366_default.diamond.piechart.jpeg" title="GridION366: Piechart for Classification Results of Diamond (default)">Figure 2</a>. No other species could be accurately classified (others: 69.076%), 12.051% of the reads are unclassified. Although the most abundant species can be identified using Diamond, the abundance does not come near the expected abundance (<a href="../attachments/tables/table13.md" title="Table 13: Abundances of classified species, Diamond, Default Database"><em><strong>Table 13</strong></em></a>).</p>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion366_default.diamond.piechart.jpeg" alt="GridION366: Piechart for Classification Results of Diamond (default)" title="GridION366: Piechart for Classification Results of Diamond (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion366_default.diamond.piechart.jpeg" title="GridION366: Piechart for Classification Results of Diamond (default)">GridION366: Piechart for Classification Results of Diamond (default)</a>. The only species with more than 1% of reads assigned to is <em>Listeria monocytogenes</em> with 18.873%. 12.051% of the reads could not be assigned to any taxa, whereas 69.076% of the reads could not be assigned on species level or the corresponding species have an abundance below 1%.</td>
</tr>
<tr>
<td style="text-align:center"></td>
<td style="text-align:left"></td>
</tr>
</tbody>
</table>
<p><em><strong>Figure 2: Classification Results for Diamond, CS Log, Default Database</strong></em> <br> <br></p>
<p>The usage of the custom database does not change the general outcome of Diamond's classification (<a href="../attachments/supplements/table_s5.md" title="Supplements, Table-S5"><em><strong>Table-S5</strong></em></a>, <a href="../attachments/supplements/figure_s4.md" title="Supplements, Figure 4"><em><strong>Figure-S4a</strong></em></a>). The same species are identified, but the abundances change slightly. <em>Listeria monocytogenes</em> has 2.889% of reads assigned, <em>Enterococcus faecalis</em> has an abundance of 1.01% and Diamond assigned 4.799% of reads to <em>Limosilactobacillus fermentum</em> (or rather <em>Lactobacillus fermentum</em>). The number of unclassified reads decreased to a total of 12.492%, but the percentage of reads assigned to other taxonomic ranks or species with less abundance than 1% increased to 78.81%. <br>
There is no classification result for GridION366 due to memory and disk space limitations.</p>
<p>An explanation for the poor results might be that Diamond's key features are pairwise alignments and frameshift alignments, the taxonomic classification is only considered as an output format. This might lead to poorer performance classifying the single reads. <br>
However, if the contribution of the metagenomic sample is not known at all, Diamond might be used on the genus level to identify the general genera present in the sample. Diamond is able to identify at least six genera that are included in the sample (abundances ranging from 1.799% to 9.495% for GridION364, default database). Considering the CS Log sample, Diamond identifies <em>Listeria</em> (59.844%) and <em>Pseudomonas</em> (4.035%). With the custom database, Diamond achieves similar abundances. <!-- Due to the relatively long runtime (Table Time) and appearing memory problems (see section "Difficulties"), no specifications and different parameters are used.--></p>
<h3 id="kaiju">Kaiju</h3>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion364_default.kaiju.piechart.jpeg" alt="GridION364: Piechart for Classification Results of Kaiju (default)" title="(a) GridION364: Piechart for Classification Results of Kaiju (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion364_default.kaiju.piechart.jpeg" title="(a) GridION364: Piechart for Classification Results of Kaiju (default)">(a) GridION364: Piechart for Classification Results of Kaiju (default)</a>. The diagram is parted into ten pieces, whereby 40.786% of reads could not be assigned to a taxon on species level, 3.883% of the reads are therefore assigned to a species but with less than 1% abundance and 14.078% of the reads are unclassified. The three most abundant species are <em>Lactobacillus fermentum</em>, <em>Enterococcus faecalis</em> and <em>Listeria monocytogenes</em> with 12.503%, 10.269% and 9.322%, respectively. The abundances of <em>Pseudomonas aeruginoa</em>, <em>Salmonella enterica</em>, <em>Escherichia coli</em> and <em>Staphylococcus aureus</em> range between 1.104% and 4.151%.</td>
</tr>
<tr>
<td style="text-align:center"><img src="../stats/pics/promethion365_default.kaiju.piechart.jpeg" alt="PromethION365: Piechart for Classification Results of Kaiju (default)" title="(b) PromethION365: Piechart for Classification Results of Kaiju (default)"></td>
<td style="text-align:left"><a href="../stats/pics/promethion365_default.kaiju.piechart.jpeg" title="(b) PromethION365: Piechart for Classification Results of Kaiju (default)">(b) PromethION365: Piechart for Classification Results of Kaiju (default)</a>. This diagram is as well parted into ten pieces, whereby 39.812% of reads could not be associated to a taxon on species level, 3.786% of the reads are therefore assigned to a species but with less than 1% abundance and 16.391% of the reads are unclassified. The three most abundant species are <em>Lactobacillus fermentum</em>, <em>Enterococcus faecalis</em> and <em>Listeria monocytogenes</em> with 12.033%, 9.956% and 9.046%, respectively. The abundances of <em>Pseudomonas aeruginosa</em>, <em>Salmonella enterica</em>, <em>Escherichia coli</em> and <em>Staphylococcus aureus</em> range between 1.111% and 3.98%.</td>
</tr>
<tr>
<td style="text-align:center"></td>
<td style="text-align:left"></td>
</tr>
</tbody>
</table>
<p><em><strong>Figure 3: Classification Results for Kaiju, CS Even, Default Database</strong></em> <br> <br></p>
<p>The other protein-based classifier used in this comparison is Kaiju. Considering the CS Even samples, Kaiju is able to identify seven of the ten species in both samples and the abundances are similar as well. 3.883% and 3.786% of reads could not be classified at all for GridION and PromethION, respectively and roughly 40% of the reads in both samples could not be assigned on the species level, therefore roughly 3.8% of the reads are assigned to species that do not reach the 1% abundance mark ([<em><strong>Table 14</strong></em>]../attachments/tables//table14.md &quot;Table 14: Abundances of classified species, Kaiju, Default Database&quot;)). The plots produced on Kaiju outputs include an entry for the percentage of reads that could not be assigned to any taxa on species level due to the way Kaiju generates report files, see, for example, <a href="../stats/pics/gridion364_default.kaiju.piechart.jpeg" title="(a) GridION364: Piechart for Classification Results of Kaiju (default)">Figure 3a</a>.
The identified species are <em>Lactobacillus fermentum</em>, <em>Enterococcus faecalis</em>,  <em>Listeria monocytogenes</em>, <em>Staphylococcus aureus</em>, <em>Escherichia coli</em>, <em>Salmonella enterica</em> and <em>Pseudomonas aeruginosa</em>. The default database does not include fungi, therefore the species that is not classified, although present in the reference database, is <em>Bacillus subtilis</em> (<a href="../stats/pics/gridion364_default.kaiju.piechart.jpeg" title="(a) GridION364: Piechart for Classification Results of Kaiju (default)">Figure 3a</a>, <a href="../stats/pics/promethion365_default.kaiju.piechart.jpeg" title="(b) PromethION365: Piechart for Classification Results of Kaiju (default)">Figure 3b</a>).</p>
<table>
<thead>
<tr>
<th>Classified species</th>
<th><em>L. monocytogenes</em></th>
<th><em>E. faecalis</em></th>
<th><em>S. aureus</em></th>
<th><em>S. enterica</em></th>
<th><em>E. coli</em></th>
<th><em>P. aeruginosa</em></th>
<th><em>L. fermentum</em></th>
<th></th>
<th>unclassified</th>
<th>others</th>
<th>different taxon level</th>
</tr>
</thead>
<tbody>
<tr>
<td>CS Even</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION364</td>
<td>9.322</td>
<td>10.269</td>
<td>4.151</td>
<td>1.779</td>
<td>2.124</td>
<td>1.104</td>
<td>12.503</td>
<td></td>
<td>14.078</td>
<td>3.883</td>
<td>40.786</td>
</tr>
<tr>
<td>PromethION365</td>
<td>9.046</td>
<td>9.956</td>
<td>3.98</td>
<td>1.784</td>
<td>2.101</td>
<td>1.111</td>
<td>12.033</td>
<td></td>
<td>16.391</td>
<td>3.786</td>
<td>39.812</td>
</tr>
<tr>
<td>CS Log</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION366</td>
<td>57.905</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td></td>
<td>11.062</td>
<td>3.389</td>
<td>27.644</td>
</tr>
<tr>
<td>PromethION367</td>
<td>58.031</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td></td>
<td>12.866</td>
<td>3.135</td>
<td>25.968</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables//table14.md" title="Table 14: Abundances of classified species, Kaiju, Default Database"><em><strong>Table 14: Abundances of classified species, Kaiju, Default Database.</strong></em></a> The table shows the classification results of Kaiju for all four samples considering the default database (in %). Note that the species that is present in the reference database but not classified is <em>Bacillus subtilis</em>. The two fungis cannot be identified with the default database, because it only includes bacterial genomes or proteomes. <br> <br></p>
<p>The CS Log samples are similar as well. Kaiju is able to associate 57.905% and 58.031% of the reads with <em>Listeria monocytogenes</em>, depending on the sequencing machine (GridION, PromethION). Roughly 11% and 12.866% of the reads are unclassified, whereas 27.644% and 25.968% could not be assigned to a taxon on the species level, respectively. Around 3% of the reads are associated with species with less than 1% abundance, see <a href="../stats/pics/gridion366_default.kaiju.piechart.jpeg" title="(a) GridION366: Piechart for Classification Results of Kaiju (default)">Figure 4a</a> and <a href="../stats/pics/promethion367_default.kaiju.piechart.jpeg" title="(b) PromethION367: Piechart for Classification Results of Kaiju (default)">Figure 4b</a>.</p>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion366_default.kaiju.piechart.jpeg" alt="GridION366: Piechart for Classification Results of Kaiju (default)" title="(a) GridION366: Piechart for Classification Results of Kaiju (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion366_default.kaiju.piechart.jpeg" title="(a) GridION366: Piechart for Classification Results of Kaiju (default)">(a) GridION366: Piechart for Classification Results of Kaiju (default)</a>. It is visible that the majority of reads is assigned to <em>Listeria monocytogenes</em> with 57.905%. Most of the remaining reads (27.644%) could not be assigned to anything on species level, therefore 3.389% of the reads are assigned to a species, but this species does not reach the abundance of 1%. 11.062% of the reads are unclassified.</td>
</tr>
<tr>
<td style="text-align:center"><img src="../stats/pics/promethion367_default.kaiju.piechart.jpeg" alt="PromethION367: Piechart for Classification Results of Kaiju (default)" title="(b) PromethION367: Piechart for Classification Results of Kaiju (default)"></td>
<td style="text-align:left"><a href="../stats/pics/promethion367_default.kaiju.piechart.jpeg" title="(b) PromethION367: Piechart for Classification Results of Kaiju (default)">(b) PromethION367: Piechart for Classification Results of Kaiju (default)</a>. It is visible that the majority of reads is assigned to <em>Listeria monocytogenes</em> with 58.031%. Most of the remaining reads (25.968%) could not be assigned to anything on species level, therefore 3.135% of the reads are assigned to a species, but this species does not reach the abundance of 1%. 12.866% of the reads are unclassified. This seems to be caused by the change in database, as no other changes are made.</td>
</tr>
<tr>
<td style="text-align:center"></td>
<td style="text-align:left"></td>
</tr>
</tbody>
</table>
<p><em><strong>Figure 4: Classification Results for Kaiju, CS Log, Default Database</strong></em> <br> <br></p>
<p>Using the custom database, Kaiju achieves poorer results (<a href="../attachments/supplements/table_s6.md" title="Supplements, Table-S6"><em><strong>Table-S6</strong></em></a>). With the default database, Kaiju is able to identify nearly all bacterial species, this is not achieved with the custom database. For GridION364, 64.834% of reads are not assigned to a species but to another taxonomic rank, whereas 3.501% are not assigned to a species that reached 1% of abundance and 12.799% of reads are unclassified (<a href="../attachments/supplements/figure_s5.md" title="Supplements, Figure 5"><em><strong>Figure-S5a</strong></em></a>). The identified species are <em>Listeria monocytogenes</em> (6.85%), <em>Limosilactobacillus fermentum</em> (5.35%). <em>Enterococcus faecalis</em> (2.371%), <em>Staphylococcus aureus</em> (1.233%), <em>Saccharomyces cerevisiae</em> (1.37%) and <em>Cryptococcus neoformans</em> (1.508%). Although not all bacterial species are identified, the fungal species can be classified this time. The results for PromethION365, the other CS Even sample, are alike with similar percentages (<a href="../attachments/supplements/figure_s5.md" title="Supplements, Figure 5"><em><strong>Figure-S5b</strong></em></a>). <br>
For CS Log samples, <em>Listeria monocytogenes</em> is again the predominant species, i.e. the only species that is identified with 42.248% of reads. 44.3% of the reads are not assigned to anything on the species level, 10.97% are unclassified and 2.481% of reads are assigned to a species that does not reach 1% of abundance. This can be observed in PromethION as well (<a href="../attachments/supplements/figure_s5.md" title="Supplements, Figure 5"><em><strong>Figure-S5c and Figure-S5d</strong></em></a>). <br>
Overall, the number of unclassified reads does not change noticeably between the use of default and custom database for the four samples, but fewer reads can be assigned on a species level.</p>
<h3 id="ccmetagen">CCMetagen</h3>
<p>CCMetagen uses KMA for aligning, which is designed to map reads against redundant databases. The tool is supposed to work for large datasets, however, no classification could be done for the deeply sequenced samples with PromethION, as discussed before. KMA throws an error stating not enough space on the device. Therefore, there is only one CS Even and one CS Log sample to analyse. Additionally, no custom database can be built.</p>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion364_default.ccmetagen.piechart.jpeg" alt="GridION364: Piechart for Classification Results of CCmetagen (default)" title="GridION364: Piechart for Classification Results of CCMetagen (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion364_default.ccmetagen.piechart.jpeg" title="GridION364: Piechart for Classification Results of CCMetagen (default)">GridION364: Piechart for Classification Results of CCmetagen (default)</a>. 26.736% of reads could not be assigned to any taxon, whereas 19.458% of reads are not assigned to a taxon on species level or the species do not reach 1% abundance. The identified species are <em>Enterococcus faecalis</em> (1.795%), <em>Escherichia coli</em> (2.069%), <em>Pseudomonas aeruginosa</em> (3.387%), <em>Limosilactobacillus fermentum</em> (9.083%), <em>Staphylococcus aureus</em> (9.837%), <em>Listeria monocytogenes</em> (11.326%) and <em>Bacillus spizizenii</em> (16.309%).</td>
</tr>
<tr>
<td style="text-align:center"></td>
<td style="text-align:left"></td>
</tr>
</tbody>
</table>
<p><em><strong>Figure 5: Classification Results for CCMetagen, CS Even, GridION, Default Database</strong></em> <br> <br></p>
<p>CCMetagen is not able to classify 26.736% of the reads of the CS Even samples and 19.458% of the reads could not be associated to a taxon on species level (<a href="../stats/pics/gridion364_default.ccmetagen.piechart.jpeg" title="GridION364: Piechart for Classification Results of CCMetagen (default)">Figure 5</a>, <a href="../attachments/tables/table15.md" title="Table 15: Abundances of classified species, CCMetagen, Default Database"><em><strong>Table 15</strong></em></a>). The remaining reads are assigned to species with abundances between 1.795% (<em>Enterococcus faecalis</em>) and 16.309% (<em>Bacillus spizizenii</em>), the number of associated reads for each species are comparably low, especially for <em>Enterococcus faecalis</em> and <em>Escherichia coli</em>. In general, CCMetagen only classified species present in the sample (<a href="../stats/pics/gridion364_default.ccmetagen.piechart.jpeg" title="GridION364: Piechart for Classification Results of CCMetagen (default)">Figure 5</a>). <em>Limosilactobacillus fermentum</em> and <em>Bacillus spizizenii</em> are, however, the new name for <em>Lactobacillus fermentum</em> and a subspecies of <em>Bacillus subtilis</em> <a href="https://doi.org/10.1007/s10482-019-01354-9" title="Dunlap, C. A., Bowman, M. J., &amp; Zeigler, D. R. (2020). Promotion of Bacillus subtilis subsp. inaquosorum, Bacillus subtilis subsp. spizizenii and Bacillus subtilis subsp. stercoris to species status. *Antonie van Leeuwenhoek*, 113(1), 1-12.">[27]</a>, respectively. The default database does not fungal genomes, but CCMetagen is not able to identify the present fungal species.</p>
<p>Considering the CS Log sample (<a href="../stats/pics/gridion366_default.ccmetagen.piechart.jpeg" title="GridION366: Piechart for Classification Results of CCMetagen (default)">Figure 6</a>, <a href="../attachments/tables/table15.md" title="Table 15: Abundances of classified species, CCMetagen, Default Database"><em><strong>Table 15</strong></em></a>) sequenced with GridION, CCMetagen is able to assign 71.494% of reads to <em>Listeria monocytogenes</em> and 2.822% to <em>Pseudomonas aeruginosa</em>. 0.279% are assigned to others and 25.405% are unclassified. <br></p>
<p>The tool achieved the highest percentage of unclassified reads.</p>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion366_default.ccmetagen.piechart.jpeg" alt="GridION366: Piechart for Classification Results of CCMetagen (default)" title="GridION366: Piechart for Classification Results of CCMetagen (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion366_default.ccmetagen.piechart.jpeg" title="GridION366: Piechart for Classification Results of CCMetagen (default)">GridION366: Piechart for Classification Results of CCMetagen (default)</a>. 26.736% of reads could not be assigned to any taxon, whereas 19.458% of reads are not assigned to a taxon on species level or the species do not reach 1% abundance. The identified species are <em>Enterococcus faecalis</em> (1.795%), <em>Escherichia coli</em> (2.069%), <em>Pseudomonas aeruginosa</em> (3.387%), <em>Limosilactobacillus fermentum</em> (9.083%), <em>Staphylococcus aureus</em> (9.837%), <em>Listeria monocytogenes</em> (11.326%) and <em>Bacillus spizizenii</em> (16.309%).</td>
</tr>
<tr>
<td style="text-align:center"></td>
<td style="text-align:left"></td>
</tr>
</tbody>
</table>
<p><em><strong>Figure 6: Classification Results for CCMetagen, CS Log, GridION, Default Database</strong></em> <br> <br></p>
<table>
<thead>
<tr>
<th>Classified Species</th>
<th><em>L. monocytogenes</em></th>
<th><em>E. faecalis</em></th>
<th><em>S. aureus</em></th>
<th><em>E. coli</em></th>
<th><em>P. aeruginosa</em></th>
<th><em>L. fermentum</em></th>
<th><em>B. spizizenii</em></th>
<th></th>
<th>unclassified</th>
<th>others</th>
</tr>
</thead>
<tbody>
<tr>
<td>CS Even</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION364</td>
<td>11.326</td>
<td>1.795</td>
<td>9.837</td>
<td>2.069</td>
<td>3.387</td>
<td>9.083</td>
<td>16.309</td>
<td></td>
<td>26.736</td>
<td>19.458</td>
</tr>
<tr>
<td>CS Log</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION366</td>
<td>71.494</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>2.822</td>
<td>-</td>
<td>-</td>
<td></td>
<td>25.405</td>
<td>0.279</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table15.md" title="Table 15: Abundances of classified species, CCMetagen, Default Database"><em><strong>Table 15: Abundances of classified species, CCMetagen, Default Database.</strong></em></a> The table shows the classification results of CCMetagen for the default database and GridION samples (in %). CCMetagen is able to identify the majority of bacterial species in the dataset and is able to identify <em>Listeria monocytogenes</em> as predominant species in the CS Log sample. However, nearly 25% of reads are unclassified in each sample. <br> <br></p>
<h3 id="centrifuge">Centrifuge</h3>
<p>For the CS Even samples, Centrifuge is able to classify 89.504% and 87.086% for GridION364 and PromethION365, respectively (<a href="../stats/pics/gridion364_default.centrifuge.piechart.jpeg" title="(a) GridION364: Piechart for Classification Results of Centrifuge (default)">Figure 7a</a>, <a href="../stats/pics/promethion365_default.centrifuge.piechart.jpeg" title="(b) PromethION365: Piechart for Classification Results of Centrifuge (default)">Figure 7b</a>). Between 3.5% and 3.9% of the reads could not be assigned to a taxon on species level.
Since the default database of Centrifuge does not contain fungal genomes, Centrifuge is only able to identify the eight bacterial species. Those species are identified with abundances ranging between 6% for <em>Salmonella enterica</em> and 18.368% for <em>Bacillus subtilis</em> for GridION and between 5.936% for <em>Salmonella enterica</em> and 17.724% for <em>Bacillus subtilis</em> for PromethION, respectively (<a href="../attachments/tables/table16.md" title="Table 16: Abundances of classified species, Centrifuge, Default Database"><em><strong>Table 16</strong></em></a>). The classification results, therefore, do not improve with greater sequencing depth. <br>
Since Centrifuge assigns a read to up to five species, the abundances are calculated slightly different: the number of reads assigned to one entry is divided by the total number of assignments which can be greater than the number of reads. <br>
The CS Log results show similarity between the samples as well. Three species can be identified with the majority of reads associated with <em>Listeria monocytogenes</em> (84.591% and 82.068% for GridIOn and PromethION, respectively). Centrifuge assigned about 1% of the reads to <em>Bacillus subtilis</em> and 4.5% to <em>Pseudomonas aeruginosa</em>. The sample sequenced with GridION shows slightly fewer unclassified reads with 7.214% in contrast to 9.649% for PromethION. There are about 2.5% reads that could not be assigned to a taxon on species level (<a href="../attachments/tables/table16.md" title="Table 16: Abundances of classified species, Centrifuge, Default Database"><em><strong>Table 16</strong></em></a>, <a href="../stats/pics/gridion366_default.centrifuge.piechart.jpeg" title="(a) GridION366: Piechart for Classification Results of Centrifuge (default)">Figure 8a</a>, <a href="../stats/pics/promethion367_default.centrifuge.piechart.jpeg" title="(b) PromethION367: Piechart for Classification Results of Centrifuge (default)">Figure 8b</a>).</p>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion364_default.centrifuge.piechart.jpeg" alt="GridION364: Piechart for Classification Results of Centrifuge (default)" title="(a) GridION364: Piechart for Classification Results of Centrifuge (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion364_default.centrifuge.piechart.jpeg" title="(a) GridION364: Piechart for Classification Results of Centrifuge (default)">(a) GridION364: Piechart for Classification Results of Centrifuge (default)</a>. Centrifuge is able to identify all bacterial species in the sample with relatively high abundances (6% to 18.368%). 10.496% of the reads could not be classified and 3.541% could not be assigned to a taxon on species level or exceed the 1% abundance mark.</td>
</tr>
<tr>
<td style="text-align:center"><img src="../stats/pics/promethion365_default.centrifuge.piechart.jpeg" alt="PromethION365: Piechart for Classification Results of Centrifuge (default)" title="(b) PromethION365: Piechart for Classification Results of Centrifuge (default)"></td>
<td style="text-align:left"><a href="../stats/pics/promethion365_default.centrifuge.piechart.jpeg" title="(b) PromethION365: Piechart for Classification Results of Centrifuge (default)">(b) PromethION365: Piechart for Classification Results of Centrifuge (default)</a>. Centrifuge is able to identify all bacterial species in the sample with relatively high abundances (5.936% to 17.724%). 12.914% of the reads could not be classified and 3.974% could not be assigned to a taxon on species level or exceed the 1% abundance mark.</td>
</tr>
<tr>
<td style="text-align:center"></td>
<td style="text-align:left"></td>
</tr>
</tbody>
</table>
<p><em><strong>Figure 7: Classification Results for Centrifuge, CS Even, Default Database</strong></em> <br> <br></p>
<table>
<thead>
<tr>
<th>Classified species</th>
<th><em>B. subtilis</em></th>
<th><em>L. monocytogenes</em></th>
<th><em>E. faecalis</em></th>
<th><em>S. aureus</em></th>
<th><em>S. enterica</em></th>
<th><em>E. coli</em></th>
<th><em>P. aeruginosa</em></th>
<th><em>L. fermentum</em></th>
<th></th>
<th>unclassified</th>
<th>other</th>
</tr>
</thead>
<tbody>
<tr>
<td>CS Even</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION364</td>
<td>18.368</td>
<td>13.514</td>
<td>11.182</td>
<td>11.244</td>
<td>6</td>
<td>5.933</td>
<td>5.21</td>
<td>14.513</td>
<td></td>
<td>10.496 (366,458)</td>
<td>3.541</td>
</tr>
<tr>
<td>PromethION365</td>
<td>17.724</td>
<td>13.018</td>
<td>10.803</td>
<td>10.782</td>
<td>5.936</td>
<td>5.886</td>
<td>5.204</td>
<td>13.759</td>
<td></td>
<td>12.914 (4,624,479)</td>
<td>3.974</td>
</tr>
<tr>
<td>CS Log</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION366</td>
<td>1.085</td>
<td>84.591</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>4.653</td>
<td>-</td>
<td></td>
<td>7.214 (264,559)</td>
<td>2.457</td>
</tr>
<tr>
<td>PromethION367</td>
<td>1.069</td>
<td>82.068</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>4.532</td>
<td>-</td>
<td></td>
<td>9.649 (3,335,852)</td>
<td>2.682</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table16.md" title="Table 16: Abundances of classified species, Centrifuge, Default Database"><em><strong>Table 16: Abundances of classified species, Centrifuge, Default Database.</strong></em></a> The table shows the classification results of Centrifuge for all four samples considering the default database (in %). The two fungi can not be identified with the default database, because it only includes microbial genomes. The column &quot;unclassified&quot; also contains the absolute number of unclassified reads.<br> <br></p>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion366_default.centrifuge.piechart.jpeg" alt="(a) GridION366: Piechart for Classification Results of Centrifuge (default)" title="(a) GridION366: Piechart for Classification Results of Centrifuge (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion366_default.centrifuge.piechart.jpeg" title="(a) GridION366: Piechart for Classification Results of Centrifuge (default)">(a) GridION366: Piechart for Classification Results of Centrifuge (default)</a>. In this sample, 8.52% of reads could not be classified at all and 5.18% of reads are assigned to taxa that are not on species level or the corresponding species has below 1% abundance. The remaining reads are assigned to <em>Listeria monocytogenes</em> (81.293%), <em>Pseudomonas aeruginosa</em> (3.98%) and <em>Bacillus subtilis</em> (1.027%)</td>
</tr>
<tr>
<td style="text-align:center"><img src="../stats/pics/promethion367_default.centrifuge.piechart.jpeg" alt="(b) PromethION367: Piechart for Classification Results of Centrifuge (default)" title="(b) PromethION367: Piechart for Classification Results of Centifuge (default)"></td>
<td style="text-align:left"><a href="../stats/pics/promethion367_default.centrifuge.piechart.jpeg" title="(b) PromethION367: Piechart for Classification Results of Centifuge (default)">(b) PromethION367: Piechart for Classification Results of Centrifuge (default)</a>. In this sample, 10.4% of reads could not be classified at all and 4.805% of reads are assigned to taxa that are not on species level or the corresponding species has below 1% abundance. The remaining reads are assigned to <em>Listeria monocytogenes</em> (79.702%), <em>Pseudomonas aeruginosa</em> (4.069%) and <em>Bacillus subtilis</em> (1.023%).</td>
</tr>
<tr>
<td style="text-align:center"></td>
<td style="text-align:left"></td>
</tr>
</tbody>
</table>
<p><em><strong>Figure 8: Classification Results for Centrifuge, CS Log, Default Database</strong></em> <br> <br></p>
<p>Using the custom database leads to slightly different results (<a href="../attachments/supplements/table_s7.md" title="Supplements, Table-S7"><em><strong>Table-S7</strong></em></a>). The main change for CS Even samples is the number of reads that are unclassified or assigned to &quot;others&quot;. The percentage of unclassified reads has decreased to 5.052%, but 20.495% of reads are not assigned to anything on species level or to species that didn't reach 1% of reads (<a href="../attachments/supplements/figure_s6.md" title="Supplements, Figure 6"><em><strong>Figure-S6a and Figure-S6b</strong></em></a>). <br>
Although fungal species are included in this database, Centrifuge is not able to assign 1% of reads to those. The remaining reads are assigned with slightly different proportions, instead of <em>Lactobacillus fermentum</em>, <em>Limosilactobacillus fermentum</em> is identified with a similar amount of reads (14.248% and 14.671%, respectively), but as discussed before, <em>Limosilactobacillus fermentum</em> is the new scientific name for <em>Lactobacillus fermentum</em>. Instead of <em>Salmonella enterica</em>, Centrifuge identifies three subspecies of <em>Salmonella</em>: <em>Salmonella sp. S048_01045</em>, <em>Salmonella sp. S102_03650</em> and <em>Salmonella sp. S060_01291</em>, which are three unclassified <em>Salmonella</em> species <a href="https://doi.org/10.1093/database/baaa062" title="Schoch, C. L., Ciufo, S., Domrachev, M., Hotton, C. L., Kannan, S., Khovanskaya, R., ... &amp; Karsch-Mizrachi, I. (2020). NCBI Taxonomy: a comprehensive update on curation, resources and tools. *Database*, 2020.">[34]</a>. Furthermore, instead of only <em>Bacillus subtilis</em>, <em>Bacillus spizizenii</em> is identified with 9.606% of reads as well, and 2.123% or reads are assigned to <em>Bacillus subtilis</em> (Default: <em>B. subtilis</em>: 18.033%). But as mentioned earlier, <em>Bacillus spizizenii</em> is a subspecies of <em>Bacillus subtilis</em>. Hence, Centrifuge is still able to identify most of the bacterial species with slight changes in abundance, but the fungal species are not identified. <br>
The CS Log samples show that the majority of reads is assigned to <em>Listeria monocytogenes</em> (366: 72.843%, 367: 70.223%), although the percentage has decreased (<a href="../attachments/supplements/table_s7.md" title="Supplements, Table-S7"><em><strong>Table-S7</strong></em></a>). This might be due to the identification of a new species - <em>Listera innouca</em>. 3.405% and 2.671% of reads are assigned to this, respectively. The percentage of reads not classified has decreased to ~6.75% and the percentage of reads assigned to &quot;others&quot; has increased to ~13% (<a href="../attachments/supplements/figure_s6.md" title="Supplements, Figure 6"><em><strong>Figure-S6c and Figure-S6d</strong></em></a>).</p>
<h3 id="clark">CLARK</h3>
<p>As for CLARK, the results of the CS Even samples are similar for GridION and PromethION. The default database does not include fungi, therefore they cannot be considered here.
Additional to the expected species, CLARK assigns 1.046% and 1.041% of the reads to <em>Bacillus vallismortis</em> for GridION and PromethION, respectively. 21.888% (22.801%) of reads are not classified, 4.068% (4.979%) could not be assigned to a taxon on species level. The bacterial species have abundances ranging from 4.562% to 14.609% and 4.66% to 13.894% (<a href="../attachments/tables/table17.md" title="Table 17: Abundances of classified species, CLARK, Default Database"><em><strong>Table 17</strong></em></a>). As seen for other classifiers, reads are assigned to <em>Limosilactobacillus fermentum</em>, instead of <em>Lactobacillus fermentum</em> due to renaming (<a href="../stats/pics/gridion364_default.clark.piechart.jpeg" title="(a) GridION364: Piechart for Classification Results of CLARK (default)">Figure 9a</a>).</p>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion364_default.clark.piechart.jpeg" alt="(a) GridION364: Piechart for Classification Results of CLARK (default)" title="(a) GridION364: Piechart for Classification Results of CLARK (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion364_default.clark.piechart.jpeg" title="(a) GridION364: Piechart for Classification Results of CLARK (default)">(a) GridION364: Piechart for Classification Results of CLARK (default)</a>. Since 25.946% of reads could not be assigned at all or to a species, the remaining reads are assigned to <em>Bacillus subtilis</em>, <em>Bacillus vallismortis</em>, <em>Enterococcus faecalis</em>,  <em>Escherichia coli</em>, <em>Limosilactobacillus fermentum</em>, <em>Listeria monocytogenes</em>, <em>Pseudomonas aeruginosa</em>, <em>Salmonella enterica</em> and <em>Staphylococcus aureus</em>.</td>
</tr>
<tr>
<td style="text-align:center"><img src="../stats/pics/promethion365_default.clark.piechart.jpeg" alt="(b) PromethION365: Piechart for Classification Results of CLARK (default)" title="(b) PromethION365: Piechart for Classification Results of CLARK (default)"></td>
<td style="text-align:left"><a href="../stats/pics/promethion365_default.clark.piechart.jpeg" title="(b) PromethION365: Piechart for Classification Results of CLARK (default)">(b) PromethION365: Piechart for Classification Results of CLARK (default)</a>. Since 27.78%% of reads could not be assigned at all or to a species, the remaining reads are assigned to <em>Bacillus subtilis</em>, <em>Bacillus vallismortis</em>, <em>Enterococcus faecalis</em>, <em>Escherichia coli</em>, <em>Limosilactobacillus fermentum</em>, <em>Listeria monocytogenes</em>, <em>Pseudomonas aeruginosa</em>, <em>Salmonella enterica</em> and <em>Staphylococcus aureus</em>.</td>
</tr>
<tr>
<td style="text-align:center"></td>
<td style="text-align:left"></td>
</tr>
</tbody>
</table>
<p><em><strong>Figure 9: Classification Results for Centrifuge, CS Even, Default Database</strong></em> <br> <br></p>
<table>
<thead>
<tr>
<th>Classified species</th>
<th><em>B. subtilis</em></th>
<th><em>L. monocytogenes</em></th>
<th><em>E. faecalis</em></th>
<th><em>S. aureus</em></th>
<th><em>S. enterica</em></th>
<th><em>E. coli</em></th>
<th><em>P. aeruginosa</em></th>
<th><em>L. fermentum</em></th>
<th><em>B. vallismortis</em></th>
<th></th>
<th>unclassified</th>
<th>other</th>
</tr>
</thead>
<tbody>
<tr>
<td>CS Even</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION364</td>
<td>7.332</td>
<td>13.242</td>
<td>11.445</td>
<td>11.208</td>
<td>5.649</td>
<td>4.961</td>
<td>4.562</td>
<td>14.609</td>
<td>1.046</td>
<td></td>
<td>21.888</td>
<td>4.068</td>
</tr>
<tr>
<td>PromethION365</td>
<td>7.219</td>
<td>12.79</td>
<td>11.064</td>
<td>10.809</td>
<td>5.651</td>
<td>5.092</td>
<td>4.66</td>
<td>13.894</td>
<td>1.041</td>
<td></td>
<td>22.801</td>
<td>4.979</td>
</tr>
<tr>
<td>CS Log</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION366</td>
<td>-</td>
<td>82.708</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>3.997</td>
<td>-</td>
<td>-</td>
<td></td>
<td>11.764</td>
<td>1.531</td>
</tr>
<tr>
<td>PromethION367</td>
<td>-</td>
<td>80.632</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>4.021</td>
<td>-</td>
<td>-</td>
<td></td>
<td>13.261</td>
<td>2.086</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table17.md" title="Table 17: Abundances of classified species, CLARK, Default Database"><em><strong>Table 17: Abundances of classified species, CLARK, Default Database.</strong></em></a> The table shows the classification results of CLARK for all four samples considering the default database (in %). The two fungis cannot be identified with the default database, because it only includes microbial genomes.<br> <br></p>
<p>CLARK is also able to identify <em>Listeria monocytogenes</em> as the most prominent species in the CS Log samples (e.g. <a href="../stats/pics/gridion366_default.clark.piechart.jpeg" title="(a) GridION366: Piechart for Classification Results of CLARK (default)">Figure 10a</a>). 82.708% and 80.632% of the reads are assigned to this species from the GridION and PromethION sample, respectively. In both samples, <em>Pseudomonas aeruginosa</em> can be identified as well with 3.997% and 4.021%, respectively. Roughly 12% of the reads could not be classified and roughly 1.75% of the reads could not be assigned to a taxon on species level (<a href="../attachments/tables/table17.md" title="Table 17: Abundances of classified species, CLARK, Default Database"><em><strong>Table 17</strong></em></a>).</p>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion366_default.clark.piechart.jpeg" alt="(a) GridION366: Piechart for Classification Results of CLARK (default)" title="(a) GridION366: Piechart for Classification Results of CLARK (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion366_default.clark.piechart.jpeg" title="(a) GridION366: Piechart for Classification Results of CLARK (default)">(a) GridION366: Piechart for Classification Results of CLARK (default)</a>. <em>Listeria monocytogenes</em> and <em>Pseudomonas aeruginosa</em> are the identified species with 82.708% of reads and 3.997%, respectively. 11.764% of reads are not classified, whereas 1.531% of reads are not assigned to a taxon on species level or the assigned species has an abundance below 1%.</td>
</tr>
<tr>
<td style="text-align:center"><img src="../stats/pics/promethion367_default.clark.piechart.jpeg" alt="(b) PromethION367: Piechart for Classification Results of CLARK (default)" title="(b) PromethION367: Piechart for Classification Results of CLARK (default)"></td>
<td style="text-align:left"><a href="../stats/pics/promethion367_default.clark.piechart.jpeg" title="(b) PromethION367: Piechart for Classification Results of CLARK (default)">(b) PromethION367: Piechart for Classification Results of CLARK (default)</a>. <em>Listeria monocytogenes</em> and <em>Pseudomonas aeruginosa</em> are the identified species with 80.632% of reads and 4.021%, respectively. 31.261% of reads are not classified, whereas 2.086% of reads are not assigned to a taxon on species level or the assigned species has an abundance below 1%.</td>
</tr>
<tr>
<td style="text-align:center"></td>
<td style="text-align:left"></td>
</tr>
</tbody>
</table>
<p><em><strong>Figure 10: Classification Results for Centrifuge, CS Log, Default Database</strong></em> <br> <br></p>
<p>The classification using the custom database has to be considered with caution due to the sparsed database, the usage of CLARK-l and therefore, the different k-mer size. Additionally, the authors state the usage of CLARK-l as a draft of the &quot;real&quot; classification that CLARK would perform. <br>
Although the majority of bacterial species of the sample are identified within the CS Even samples except for <em>Bacillus subtilis</em> and <em>Listeria monocytogenes</em>, the classification results vary from the corresponding outcomes using the default database (<a href="../attachments/supplements/table_s8.md" title="Supplements, Table-S8"><em><strong>Table-S8</strong></em></a>). <em>Escherichia coli</em> has a greater amount of reads associated with, with up to 22.5% (default: ~5%). There are various new species with abundances ranging from 1% to 15%. Around 15% of reads are assigned to <em>Lactiplantibacillus plantarum</em> which is previously known as <em>Lactobacillus plantarum</em>, which could explain the number of reads assigned to it, since <em>Lactobacillus fermentum</em> is a species in the dataset. This is true for GridION and PromethION. Other species, however, do not have a connection to species of the dataset like <em>Thermoascus verrucosus</em>, which is a fungus, or <em>Klebsiella pneumoniae</em> (<a href="../attachments/supplements/figure_s7.md" title="Supplements, Figure 7"><em><strong>Figure-S7a and Figure-S7b</strong></em></a>). The fungal species of the dataset, however, are not identified with more than 1% of reads. Compared to the default database, the number of unclassified reads decreased for all samples (GridION364, for example; default: 21.888%, custom: 14.407%), but the number of reads assigned to &quot;others&quot; decreased from 4.068% to 15.863% for GridION364, default and custom database, respectively. This is also true for PromethION365. <br>
Considering the CS Log samples, the number of unclassified reads increased to ~20% (<a href="../attachments/supplements/table_s8.md" title="Supplements, Table-S8"><em><strong>Table-S8</strong></em></a>) instead of ~12% (<a href="../attachments/tables/table17.md" title="Table 17: Abundances of classified species, CLARK, Default Database"><em><strong>Table 17</strong></em></a>), the percentage of reads assigned to &quot;others&quot; increased drastically as well. The classification results of these samples vary a lot from those using the default database. Instead of a predominant assignment to <em>Listeria monocytogenes</em> (which still has a fifth of reads assigned to in both samples), there are several species with small amounts of reads ranging between 1% to 6% including <em>Escherichia coli</em>, <em>Salmonella enterica</em> and <em>Staphylococcus aureus</em>, but also species like <em>Klebsiella penumoniae</em> and <em>Kocuria rhizophila</em>, which do not belong to the species of the dataset (<a href="../attachments/supplements/figure_s7.md" title="Supplements, Figure 7">***Figure-S7c and Figure-S7d</a>). <br>
The varying results might be explained with the database and k-mer size. A shorter k-mer of 27 nt instead of 31 nt (used for default database) might lead to wider classification hits and therefore various species are identifies, especially close species, as might be the case with <em>Lactiplantibacillus plantarum</em>. On the other hand, the shorter k-mers should lead to higher sensitivity. The sparse database/index is not expected to have the quality of a full database, which contributes to wider classification results. However, these classification results should be handled as a draft and for this, they seem to display the general composition of the CS Even and CS Log sample, even if there are several additional species and the abundances vary.</p>
<h3 id="kraken2">Kraken2</h3>
<p>The Kraken2 default database includes the fungal genomes, therefore this is the only classifier that has the possibility to accurately identify all present species using the default database. Both, the GridION and PromethION sequenced samples,  identify all ten species and no other species different to the ground truth are identified. The read abundances range from 2.003% to 17.514% for GridION364 and 1.954% to 16.991% for PromethION365. On average for the CS Even samples, 10% of reads could not be classified at all, and 4.5% are not assigned on species level (<a href="../stats/pics/gridion364_default.kraken2.piechart.jpeg" title="(a) GridION364: Piechart for Classification Results of Kraken2 (default)">Figure 11a</a>, <a href="../stats/pics/promethion365_default.kraken2.piechart.jpeg" title="(b) PromethION365: Piechart for Classification Results of Kraken2 (default)">Figure 11b</a>, <a href="../attachments/tables/table18.md" title="Table 18: Abundances of classified species, Kraken2, Default Database"><em><strong>Table 18</strong></em></a>).</p>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion364_default.kraken2.piechart.jpeg" alt="(a) GridION364: Piechart for Classification Results of Kraken2 (default)" title="(a) GridION364: Piechart for Classification Results of Kraken2 (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion364_default.kraken2.piechart.jpeg" title="(a) GridION364: Piechart for Classification Results of Kraken2 (default)">(a) GridION364: Piechart for Classification Results of Kraken2 (default)</a>. Kraken2 identified the species present in the CS Even samples, the abundances range from 2.003% for <em>Cryptococcus neoformans</em> to 17.514% for <em>Bacillus subtilis</em>. 9.13% of the reads are unclassified and 4.566% of the reads are assigned to a taxon on a different level (not species) or the corresponding species has no abundance of 1%.</td>
</tr>
<tr>
<td style="text-align:center"><img src="../stats/pics/promethion365_default.kraken2.piechart.jpeg" alt="(b) PromethION365: Piechart for Classification Results of Kraken2 (default)" title="(b) PromethION365: Piechart for Classification Results of Kraken2 (default)"></td>
<td style="text-align:left"><a href="../stats/pics/promethion365_default.kraken2.piechart.jpeg" title="(b) PromethION365: Piechart for Classification Results of Kraken2 (default)">(b) PromethION365: Piechart for Classification Results of Kraken2 (default)</a>. Kraken2 identified the species present in the CS Even samples, the abundances range from 1.954% for <em>Cryptococcus neoformans</em> to 16.991% for <em>Bacillus subtilis</em>. 11.42% of the reads are unclassified and 4.569% of the reads are assigned to a taxon on a different level (not species) or the corresponding species has no abundance of 1%.</td>
</tr>
<tr>
<td style="text-align:center"></td>
<td style="text-align:left"></td>
</tr>
</tbody>
</table>
<p><em><strong>Figure 11: Classification Results for Kraken2, CS Even, Default Database</strong></em> <br> <br></p>
<table>
<thead>
<tr>
<th>Classified Species</th>
<th><em>B. subtilis</em></th>
<th><em>L. monocytogenes</em></th>
<th><em>E. faecalis</em></th>
<th><em>S. aureus</em></th>
<th><em>S. enterica</em></th>
<th><em>E. coli</em></th>
<th><em>P. aeruginosa</em></th>
<th><em>L. fermentum</em></th>
<th><em>S. cerevisiae</em></th>
<th><em>C. neoformans</em></th>
<th></th>
<th>unclassified</th>
<th>others</th>
</tr>
</thead>
<tbody>
<tr>
<td>CS Even</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION364</td>
<td>17.514</td>
<td>12.87</td>
<td>11.111</td>
<td>11.067</td>
<td>5.685</td>
<td>5.255</td>
<td>4.481</td>
<td>14.142</td>
<td>2.176</td>
<td>2.003</td>
<td></td>
<td>9.13</td>
<td>4.566</td>
</tr>
<tr>
<td>PrometION365</td>
<td>16.991</td>
<td>12.455</td>
<td>10.757</td>
<td>10.676</td>
<td>5.703</td>
<td>5.299</td>
<td>4.612</td>
<td>12.477</td>
<td>2.096</td>
<td>1.954</td>
<td></td>
<td>11.42</td>
<td>4.559</td>
</tr>
<tr>
<td>CS Log</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION366</td>
<td>1.027</td>
<td>81.293</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>3.98</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td></td>
<td>8.52</td>
<td>5.18</td>
</tr>
<tr>
<td>PromethION367</td>
<td>1.023</td>
<td>79.702</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>4.069</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td></td>
<td>10.4</td>
<td>4.805</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table18.md" title="Table 18: Abundances of classified species, Kraken2, Default Database"><em><strong>Table 18: Abundances of classified species, Kraken2, Default Database.</strong></em></a> Kraken2 is able to assign the majority of reads. Over all four samples, an average of 9.87% of reads are unclassified, the PromethION samples showing a slightly increased rate of unclassified reads. With the CS Even samples, all ten species are identified. Considering the CS Log samples, <em>Listeria monocytogenes</em> is identified as a predominant species with 81.283% and 79.70% for GridIOnN and PromethION, respectively. The two other species with at least 1% abundance are <em>Bacillus subtilis</em> and <em>Pseudomonas aeruginosa</em>. <br> <br></p>
<p>The classification results for CS Log are similar between the sequencing methods as well. The majority of reads is assigned to <em>Listeria monocytogenes</em> (81.293% and 79.702%, respectively). The other two identified species are <em>Pseudomonas aeruginosa</em> and <em>Bacillus subtilis</em> with roughly 4% and 1%, respectively (<a href="../stats/pics/gridion366_default.kraken2.piechart.jpeg" title="(a) GridION366: Piechart for Classification Results of Kraken2 (default)">Figure 12a</a>, <a href="../stats/pics/promethion367_default.kraken2.piechart.jpeg" title="(b) PromethION367: Piechart for Classification Results of Kraken2 (default)">Figure 12b</a>).</p>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion366_default.kraken2.piechart.jpeg" alt="(a) GridION366: Piechart for Classification Results of Kraken2 (default)" title="(a) GridION366: Piechart for Classification Results of Kraken2 (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion366_default.kraken2.piechart.jpeg" title="(a) GridION366: Piechart for Classification Results of Kraken2 (default)">(a) GridION366: Piechart for Classification Results of Kraken2 (default)</a>. Kraken2 is able to assign the majoity of reads to <em>Listeria monocytogenes</em> with 81.293%. 1.027% are assigned to <em>Bacillus subtilis</em> and 3.98% are assigned to <em>Pseudomonas aeruginosa</em>. 8.52% of reads are not classified at all. The percentage of reads that are classified to a taxon not on species level or a species with less than 1% abundance is 5.18%.</td>
</tr>
<tr>
<td style="text-align:center"><img src="../stats/pics/promethion367_default.kraken2.piechart.jpeg" alt="(b) PromethION367: Piechart for Classification Results of Kraken2 (default)" title="(b) PromethION367: Piechart for Classification Results of Kraken2 (default)"></td>
<td style="text-align:left"><a href="../stats/pics/promethion367_default.kraken2.piechart.jpeg" title="(b) PromethION367: Piechart for Classification Results of Kraken2 (default)">(b) PromethION367: Piechart for Classification Results of Kraken2 (default)</a>. Kraken2 is able to assign the majoity of reads to <em>Listeria monocytogenes</em> with 79.702%. 1.023% are assigned to <em>Bacillus subtilis</em> and 4.069% are assigned to <em>Pseudomonas aeruginosa</em>. 10.4% of reads are not classified at all. The percentage of reads that are classified to a taxon not on species level or a species with less than 1% abundance is 4.805%.</td>
</tr>
<tr>
<td style="text-align:center"></td>
<td style="text-align:left"></td>
</tr>
</tbody>
</table>
<p><em><strong>Figure 12: Classification Results for Kraken2, CS Log, Default Database</strong></em> <br> <br></p>
<p>Kraken2 reached similar classification results with the the custom database (<a href="../attachments/supplements/table_s9.md" title="Supplements, Table-S9"><em><strong>Table-S9</strong></em></a>). <br>
The main difference is the percentage of reads associated with <em>Bacillus subtilis</em>, which decreased to roughly 5% for the CS Even samples. However, about 10.5% of the reads are instead identified as <em>Bacillus spizizenii</em>, which is, as already mentioned, a subspecies of <em>Bacillus subtilis</em>. Taking the percentages for both species adds up roughly to the amount of reads Kraken2 assigns to <em>Bacillus subtilis</em> using the default database. The fungal species are identified with about 2% each in each sample (<a href="../attachments/supplements/figure_s8.md" title="Supplements, Figure 8"><em><strong>Figure-S8a and Figure-S8b</strong></em></a>).<br>
The percentage of unclassified reads is similar (9.56% and 11.76%, respectively), but the percentage of reads assigned to &quot;others&quot; increased to ~8%. <br>
Considering the CS Log samples, the results are similar as well. The majority of reads is identified as <em>Listeria monocytogenes</em> with 81.607% and 80.044% for GridION (<a href="../attachments/supplements/figure_s8.md" title="Supplements, Figure 8"><em><strong>Figure-S8c and Figure-S8c</strong></em></a>), respectively. Around 4.5% of reads are assigned to <em>Pseudomonas aeruginosa</em>. The number of reads that are not classified as well as the number of reads associated with taxa that are not on species level or to species that does not reach 1% abundance are similar for both databases.</p>
<h3 id="bugseq">BugSeq</h3>
<p>Classification with BugSeq is done with their BugSeq Default database for the GridION samples. The classification results are stored <a href="../attachment/bugseq" title="Results for BugSeq, GridION364 and GridION366">here</a>. <br>
BugSeq is able to identify all species present in the dataset with abundances similar to the estimated ones. This is true for the CS Even and CS Log samples (<a href="../attachments/tables/table19.md" title="Table 19: Abundances of classified species, BugSeq, Default Database"><em><strong>Table 19</strong></em></a>). <br>
Overall, 2.889% of reads are unclassified and 3.237% of the reads are assigned to a different taxonomic rank or species with less than 1% abundance considering GridION364 (<a href="../stats/pics/gridion364_default.bugseq.piechart.jpeg" title="(a) GridION364: Piechart for Classification Results of BugSeq (default)"><em><strong>Figure 13a</strong></em></a>). The fungal species are identified with around 2$ of reads each, the abundances of the bacterial species range from 5.531% (<em>Pseudomonas aeruginosa</em>) to 18.881% (<em>Bacillus subtilis</em>), which, as mentioned, equals the abundances that are estimated by Nichholls et al. in 2019 <a href="https://doi.org/10.1093/gigascience/giz043" title="Nicholls, S. M., Quick, J. C., Tang, S., &amp; Loman, N. J. (2019). Ultra-deep, long-read nanopore sequencing of mock microbial community standards. *Gigascience*, 8(5), giz043.">[14]</a>, <a href="../attachments/tables/table1.md" title="Table 1: General Information about the Used Species"><em><strong>Table 1</strong></em></a>. <br>
The classification of the CS Even samples GridION366 shows a majority of reads assigned to <em>Listeria monocytogenes</em> with 88.053%. The other two species that are identified are <em>Bacillus subtilis</em> and <em>Pseudomonas aeruginosa</em>, but the abundances are lower with 1.087% and 4.913%. Although <em>Saccharomyces cerevisiae</em> is supposed to have the same abundance, BugSeq is not able to identify the species like <em>Bacillus subtilis</em> (<a href="../stats/pics/gridion366_default.bugseq.piechart.jpeg" title="(b) GridION366: Piechart for Classification Results of BugSeq (default)"><em><strong>Figure 13b</strong></em></a>). 2.494% of reads are unclassified and 3.454% of the reads are associated with a different taxonomic rank or assigned to a species that does not reach 1% abundance.</p>
<table>
<thead>
<tr>
<th style="text-align:center"></th>
<th style="text-align:left"></th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion364_default.bugseq.piechart.jpeg" alt="(a) GridION364: Piechart for Classification Results of BugSeq (default)" title="(a) GridION364: Piechart for Classification Results of BugSeq (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion364_default.bugseq.piechart.jpeg" title="(a) GridION364: Piechart for Classification Results of BugSeq (default)">(a) GridION364: Piechart for Classification Results of BugSeq (default)</a>. The classification with BugSeq results in the identification of the species present in the CS Even dataset with abundances that correspond to the estimated values shown in <a href="../attachments/tables/table1.md" title="Table 1: General Information about the Used Species"><em><strong>Table 1</strong></em></a>. This means the fungal species have around 2% of reads associated with, whereas the range of abundance is higher for the bacterial species: From 5.531% to 18.881%. The five species with the majority of reads assigned to are <em>Bacillus subtilis</em> (18.881%), <em>Lactobacillus fermentum</em> (15.202%), <em>Listeria monocytogenes</em> (13.743%), and <em>Staphylococcus aureus</em> (11.678%). 2.889% of reads are unclassified whereas are assigned to a different taxonomic rank or to a species that does not reach 1% of abundance.</td>
</tr>
<tr>
<td style="text-align:center"><img src="../stats/pics/gridion366_default.bugseq.piechart.jpeg" alt="(b) GridION366: Piechart for Classification Results of BugSeq (default)" title="(b) GridION366: Piechart for Classification Results of BugSeq (default)"></td>
<td style="text-align:left"><a href="../stats/pics/gridion366_default.bugseq.piechart.jpeg" title="(b) GridION366: Piechart for Classification Results of BugSeq (default)">(b) GridION366: Piechart for Classification Results of BugSeq (default)</a>. For the CS Log samples classified with BugSeq, the results also correspond to the estimated abundances (<a href="../attachments/tables/table1.md" title="Table 1: General Information about the Used Species"><em><strong>Table 1</strong></em></a>). The majority of reads is assigned to <em>Listeria monocytogenes</em> with 88.053%. Two other species are identified with more than 1% of abundance: <em>Bacillus subtilis</em> and <em>Pseudomonas aeruginosa</em> with 1.087% and 4.913% respectively. The amount of unclassified reads is 2.494% and 3.454% of the reads are associated with other taxonomic ranks or species that do not reach 1% of abundance.</td>
</tr>
<tr>
<td style="text-align:center"></td>
<td style="text-align:left"></td>
</tr>
</tbody>
</table>
<p><em><strong>Figure 13: Classification Results for BugSeq, CS Even, Default Database</strong></em></p>
<table>
<thead>
<tr>
<th>Classified Species</th>
<th><em>B. subtilis</em></th>
<th><em>L. monocytogenes</em></th>
<th><em>E. faecalis</em></th>
<th><em>S. aureus</em></th>
<th><em>E. coli</em></th>
<th><em>P. aeruginosa</em></th>
<th><em>L. fermentum</em></th>
<th><em>S. cerevisiae</em></th>
<th><em>C. neoformans</em></th>
<th></th>
<th>unclassified</th>
<th>others</th>
</tr>
</thead>
<tbody>
<tr>
<td>CS Even</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION364</td>
<td>18.881</td>
<td>13.743</td>
<td>12.003</td>
<td>11.678</td>
<td>6.267</td>
<td>5.531</td>
<td>15.202</td>
<td>2.362</td>
<td>1.957</td>
<td></td>
<td>2.889</td>
<td>3.237</td>
</tr>
<tr>
<td>CS Log</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>GridION366</td>
<td>1.087</td>
<td>88.053</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td>4.913</td>
<td>-</td>
<td>-</td>
<td>-</td>
<td></td>
<td>2.494</td>
<td>3.454</td>
</tr>
</tbody>
</table>
<p><a href="../attachments/tables/table19.md" title="Table 19: Abundances of classified species, BugSeq, Default Database"><em><strong>Table 19: Abundances of classified species, BugSeq, Default Database.</strong></em></a> This table shows the calculated abundances for the GridION samples using BugSeq (in %). It can be seen that the calculated abundances correspond to the estimated values. This is true for the CS Even as well as the CS Log sample.</p>
<h2 id="difficulties-with-the-tools">Difficulties with the Tools</h2>
<p>At the beginning of this comparison, more than 12 tools should be evaluated. Some of those tools are specifically for long-read data, others not. Additional to the difficulties which lead to not using some tools which will be discussed in this section, the commands used (or tried to use) for installation and setup can be seen <a href="../attachments/commands.md" title="Installation and Preparation Commands">here</a>. <br></p>
<p>The Naive Bayes Classification Tool <a href="https://academic.oup.com/bioinformatics/article/27/1/127/202209" title="Rosen, G. L., Reichenberger, E. R., &amp; Rosenfeld, A. M. (2011). NBC: the Naive Bayes Classification tool webserver for taxonomic classification of metagenomic reads. *Bioinformatics*, 27(1), 127-129.">[28]</a> should be used to test whether older classification approaches are able to handle the state-of-the-art long reads, but the <a href="http://nbc.ece.drexel.edu/">webserver</a> is only able to process 20 Mb files for fungal samples. Trying to only use the bacterial database failed during the upload (GridiION364), the page did not respond. It seems that this tool is not able to process high-throughput data. <br></p>
<p>K-SLAM <a href="https://academic.oup.com/nar/article/45/4/1649/2674183" title="Ainsworth, D., Sternberg, M. J., Raczy, C., &amp; Butcher, S. A. (2017). k-SLAM: accurate and ultra-fast taxonomic classification and gene identification for large metagenomic data sets. *Nucleic acids research*, 45(4), 1649-1656.">[29]</a> claims to be able to process up to 10,000,000 reads, but repeats to throw a bad alloc error (<a href="../attachments/kslam.error.log">see here</a>). <br>
After trying different <tt>--num-reads-at-once</tt> values, the tool run several weeks with <tt>--num-reads-at-once 1000000</tt> without any visible progress or output. It was terminated and decided not to be used. <br></p>
<p>The lightweight alignment-free and assembly-free framework for metagenomic classification tool LiME <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03628-w" title="Guerrini, V., Louza, F. A., &amp; Rosone, G. (2020). Metagenomic analysis through the extended Burrows-Wheeler transform. *BMC bioinformatics*, 21(8), 1-25.">[30]</a> could not be installed via conda and the manual installation with git caused several errors while trying to install dependencies. It could not be installed and therefore, it is not used. <br></p>
<p>taxMaps <a href="https://genome.cshlp.org/content/28/5/751" title="Corvelo, A., CLARKe, W. E., Robine, N., &amp; Zody, M. C. (2018). taxMaps: comprehensive and highly accurate taxonomic classification of short-read data in a reasonable time. *Genome research*, 28(5), 751-758.">[31]</a> is a Python-based classification tool for short-read data. Trying to use it for long-read data resulted in several Python IndexErrors. <br></p>
<p>MetaOthello <a href="https://doi.org/10.1093/bioinformatics/btx432" title="Liu, X., Yu, Y., Liu, J., Elliott, C. F., Qian, C., &amp; Liu, J. (2018). A novel data structure to support the ultra-fast taxonomic classification of metagenomic sequences with k-mer signatures. *Bioinformatics*, 34(1), 171-178.">[32]</a> is a probabilistic hashing classifier for metagenomic reads. During the processing of a GridION sample, the tool repeatedly throws a <tt>Segmentation Fraud</tt>. It, therefore, was decided not to use the tool. <br></p>
<p>DeepMicrobes <a href="https://doi.org/10.1093/nargab/lqaa009" title="Liang, Q., Bible, P. W., Liu, Y., Zou, B., &amp; Wei, L. (2020). DeepMicrobes: taxonomic classification for metagenomics with deep learning. *NAR Genomics and Bioinformatics*, 2(1), lqaa009.">[33]</a> uses deep learning for taxonomic classification of metagenomes. The classification preprocessing and the classification itself consists of several steps and minimizes usability. The first step of the analysis is to convert the fastq sequences to TFRecord-format, this is needed for training. This step failed several times due to various reasons. However, the classification itself needs converted files as well, therefore this tool could not be used. <br></p>
<!--BugSeq [[25]](https://doi.org/10.1186/s12859-021-04089-5 "Fan, J., Huang, S., & Chorlton, S. D. (2021). BugSeq: a highly accurate cloud platform for long-read metagenomic analyses. *BMC bioinformatics*, 22(1), 1-12.") is a cloud-based pipeline for classification, therefore no parameters can be specified. The website can be reached with [https://bugseq.com/free](https://bugseq.com/free). However, the implemented tools are used with default parameters. <br>
First, a quality control with fastp is performed, the reads are then mapped with minimap2 against RefSeq sequences of the corresponding database. Pathoscope reassigns the alignments to the reference sequences using a bayesian statistical framework. The reads are classified based on the lowest common ancestor [[25]](https://doi.org/10.1186/s12859-021-04089-5 "Fan, J., Huang, S., & Chorlton, S. D. (2021). BugSeq: a highly accurate cloud platform for long-read metagenomic analyses. *BMC bioinformatics*, 22(1), 1-12.").  <br>
The advanced options consist of the choice of the metagenomic database. The options are either the 'BugSeq Default', which includes all bacterial genomes in RefSeq as well as fungal, protozoal and viral genomes in RefSeq, the human genome and transcriptome are included as well. The completion status is not taken into account. The other database is the NCBI nt, but for this analysis, it is not chosen but the 'BugSeq Default' database. <br>
The other option is a specification of the sample type. Since this dataset includes two fungal sequences, the option <tt>bacterial isolates</tt> could not be used, therefore the default <tt>Generic</tt> sample type has been chosen. The tool is avaiable in two options: academic and lab. The academic version allows only for uploads up to 10 Gb, whereas the Lab version involved monthly payment. It is currently not possible to try out the Lab version if there already exists an account. This means, no samples could be submitted since the smallest sample has a size of at least 11 Gb. -->
<p>CCMetagen and Diamond are two tools used within this comparison, but only for the samples with up to 3.5 million reads. The PromethION samples could not be classified using these tools in a reasonable time. Diamond used up to two weeks for a PromethION sample and stopped without reason or result. Due to this, it was decided to only use Diamond for the GridION samples. **With the custom database, another limitation occurred for Diamond: The use of sample GridION366 with the custom database seems to exceed the memory and disk space limitations, several runs are aborted. The variation of the block size parameter does not change that. Some error messages can be seen <a href="../attachments/gridion366_custom.diamond.error">here</a>. Since CCMetagen is based on KMA, the failed preprocessing with KMA while finding k-mer ankers lead to the use of CCMetagen only for GridION samples as well. Additionally, it is not possible to build a KMA index for the custom database, even in sparse mode, the computing takes too much memory and is aborted several times.<br>
For CLARK, the database and index creation with the custom sequences is only possible using the sparse mode and CLARK-l, which generates a sparse database and the k-mer indices have a length of 27 nt than the default 31 nt. <br></p>
<p>BugSeq achieves good results for the dataset, but the cloud service and therefore the websites still has some bugs incorporated. It might, for example, not be possible to delete the account or change the password. Additionally, a 10 Gb limit for filesize with the academic version might be a limitation that is not suitable for metagenomic samples.</p>
<h1 id="conclusion">Conclusion</h1>
<p>This research aimed to benchmark and compare different classification tools regarding their usability for metagenomic long-read data. For this, four nanopore samples based on the Zymo Community Standards are analysed using six tools mostly successful (Diamond, Kaiju, CCmetagen, Centrifuge, CLARK, Kraken2) and evaluation of the outcomes is done using common metrics like Area Under Precision-Recall Curve, Abundance Profile Similarities as well as runtime consumption. For this purpose, the generated output files are formated into a universal format which is then used to process the data regarding the metrics. Additionally, the difficulties of several tools are discussed.<br>
The evaluation revealed that Kraken2, Centrifuge and CLARK promise the best results for this dataset. At least for the used default database, the tools achieve good results considering the metrics and classification results themselves. Although there is deterioration with the custom database, those tools still achieve comparably good results. CLARK, however, might have trouble with building the needed indices due to memory limitations. CCMetagen is able to identify the present species, but the metrics show poorer results than the other DNA-based tools. A database building for the custom database is not possible. Regarding the protein-based tools, Kaiju performs better than Diamond and achieves, for the custom database, similar results than Centrifuge and Kraken2. Diamond is not able to identify the species in the dataset with appropriate abundances. <br>
It is possible that BugSeq might become a valuable tool for classification once the smaller bugs are fixed. <br>
The validation using Multi Locus Sequence Types showed the expected results and it is able to assign the gene fragments to reads in the sample. <br>
It has to be considered that the scope of this project is rather small due to the limited number of samples and databases as well as the number of successfully working tools. Therefore, a similar comparison but with several datasets might ensure more reliable and accurate outcomes regarding the metrics.</p>
<h1 id="attachments-and-supplementary-information">Attachments and Supplementary Information</h1>
<h2 id="tables">Tables</h2>
<p><a href="../attachments/supplements/table_s1.md" title="Supplements, Table-S1"><em><strong>Table-S1: Overview of abundances for GridION364, Default Database</strong></em></a></p>
<p><a href="../attachments/supplements/table_s2.md" title="Supplements, Table-S2"><em><strong>Table-S2: Overview of abundances for PromethION365, Default Database</strong></em></a></p>
<p><a href="../attachments/supplements/table_s3.md" title="Supplements, Table-S3"><em><strong>Table-S3: Overview of abundances for GridION366, Default Database</strong></em></a></p>
<p><a href="../attachments/supplements/table_s4.md" title="Supplements, Table-S4"><em><strong>Table-S4: Overview of abundances for PromethION367, Default Database</strong></em></a></p>
<p><a href="../attachments/supplements/table_s5.md" title="Supplements, Table-S5"><em><strong>Table-S5: Abundances of classified species, Diamond (custom)</strong></em></a></p>
<p><a href="../attachments/supplements/table_s6.md" title="Supplements, Table-S6"><em><strong>Table-S6: Abundances of classified species, Kaiju, (custom)</strong></em></a></p>
<p><a href="../attachments/supplements/table_s7.md" title="Supplements, Table-S7"><em><strong>Table-S7: Abundances of classified species, Centrifuge (custom)</strong></em></a></p>
<p><a href="../attachments/supplements/table_s8.md" title="Supplements, Table-S8"><em><strong>Table-S8: Abundances of classified species, CLARK (custom)</strong></em></a></p>
<p><a href="../attachments/supplements/table_s9.md" title="Supplements, Table-S9"><em><strong>Table-S9: Abundances of classified species, Kraken2 (custom)</strong></em></a></p>
<p><a href="../attachments/supplements/table_s10.md" title="Supplements, Table-S10"><em><strong>Table-S10: Overview of abundances for GridION364, Custom Database</strong></em></a></p>
<p><a href="../attachments/supplements/table_s11.md" title="Supplements, Table-S11"><em><strong>Table-S11: Overview of abundances for PromethION365, Custom Database</strong></em></a></p>
<p><a href="../attachments/supplements/table_s12.md" title="Supplements, Table-S12"><em><strong>Table-S12: Overview of abundances for GridION366, Custom Database</strong></em></a></p>
<p><a href="../attachments/supplements/table_s13.md" title="Supplements, Table-S13"><em><strong>Table-S13: Overview of abundances for PromethION367, Custom Database</strong></em></a></p>
<h2 id="figures">Figures</h2>
<p><a href="../attachments/supplements/figure_s1.md" title="Supplements, Figure 1"><em><strong>Figure-S1: GridION364: PR Curve, Diamond (default), Python</strong></em></a></p>
<p><a href="../attachments/supplements/figure_s2.md" title="Supplements, Figure 2"><em><strong>Figure-S2: GridION364: PR Curve for Diamond and Kaiju (default), no Fungi</strong></em></a></p>
<p><a href="../attachments/supplements/figure_s3.md" title="Supplements, Figure 3"><em><strong>Figure-S3: Piecharts of Classification Results, Diamond, Default Database, Genus Level</strong></em></a></p>
<p><a href="../attachments/supplements/figure_s4.md" title="Supplements, Figure 4"><em><strong>Figure-S4: Classification Results of Diamond, Custom Database, GridION364</strong></em></a></p>
<p><a href="../attachments/supplements/figure_s5.md" title="Supplements, Figure 5"><em><strong>Figure-S5: Classification results of Kaiju, Custom Database</strong></em></a></p>
<p><a href="../attachments/supplements/figure_s6.md" title="Supplements, Figure 6"><em><strong>Figure-S6: Classification Results for Centrifuge, Custom Database</strong></em></a></p>
<p><a href="../attachments/supplements/figure_s7.md" title="Supplements, Figure 7"><em><strong>Figure-S7: Classification Results for CLARK, Custom Database</strong></em></a></p>
<p><a href="../attachments/supplements/figure_s8.md" title="Supplements, Figure 8"><em><strong>Figure-S8: Classification Results for Kraken2, Custom Database</strong></em></a></p>
<p><a href="../attachments/supplements/figure_s9.md" title="Supplements, Figure 9"><em><strong>Figure-S9: Overview of the different Precision Recall Curves, Default Database</strong></em></a></p>
<p><a href="../attachments/supplements/figure_s10.md" title="Supplements, Figure 10"><em><strong>Figure-S10: Overview of the different Precision Recall Curves, Custom Database</strong></em></a></p>
<h1 id="citations">Citations</h1>
<p><a href="https://doi.org/10.1093/bib/bbs035">[1]</a> Li, W., Fu, L., Niu, B., Wu, S., &amp; Wooley, J. (2012). Ultrafast clustering algorithms for metagenomic sequence analysis. <em>Briefings in bioinformatics</em>, 13(6), 656-668.</p>
<p><a href="https://doi.org/10.1007/s10311-020-01010-z">[2]</a> Datta, S., Rajnish, K. N., Samuel, M. S., Pugazlendhi, A., &amp; Selvarajan, E. (2020). Metagenomic applications in microbial diversity, bioremediation, pollution monitoring, enzyme and drug discovery. A review. <em>Environmental Chemistry Letters</em>, 18(4), 1229-1241.</p>
<p><a href="https://doi.org/10.1016/j.gpb.2016.05.004">[3]</a> Lu, H., Giordano, F., &amp; Ning, Z. (2016). Oxford Nanopore MinION sequencing and genome assembly. <em>Genomics, proteomics &amp; bioinformatics</em>, 14(5), 265-279.</p>
<p><a href="https://doi.org/10.1186/s13059-019-1891-0">[4]</a> Wood, D. E., Lu, J., &amp; Langmead, B. (2019). Improved metagenomic analysis with Kraken 2. <em>Genome biology</em>, 20(1), 1-13.</p>
<p><a href="https://doi.org/10.1186/s12864-015-1419-2">[5]</a> Ounit, R., Wanamaker, S., Close, T. J., &amp; Lonardi, S. (2015). CLARK: fast and accurate classification of metagenomic and genomic sequences using discriminative k-mers. <em>BMC genomics</em>, 16(1), 1-13.</p>
<p><a href="https://doi.org/10.1186/s13059-020-02014-2">[6]</a> Marcelino, V. R., Clausen, P. T., Buchmann, J. P., Wille, M., Iredell, J. R., Meyer, W., ... &amp; Holmes, E. C. (2020). CCMetagen: comprehensive and accurate identification of eukaryotes and prokaryotes in metagenomic data. <em>Genome biology</em>, 21, 1-15.</p>
<p><a href="https://doi.org/10.1101%2Fgr.210641.116">[7]</a> Kim, D., Song, L., Breitwieser, F. P., &amp; Salzberg, S. L. (2016). Centrifuge: rapid and sensitive classification of metagenomic sequences. <em>Genome research</em>, 26(12), 1721-1729.</p>
<p><a href="https://doi.org/10.1186/s13062-018-0208-7">[8]</a> Menzel, P., Ng, K. L., &amp; Krogh, A. (2016). Fast and sensitive taxonomic classification for metagenomics with Kaiju. <em>Nature communications</em>, 7(1), 1-9.</p>
<p><a href="https://doi.org/10.1038/nmeth.3176">[9]</a> Buchfink, B., Xie, C., &amp; Huson, D. H. (2015). Fast and sensitive protein alignment using DIAMOND. <em>Nature methods</em>, 12(1), 59-60.</p>
<p><a href="https://doi.org/10.1016/j.cell.2019.07.010">[10]</a> Simon, H. Y., Siddle, K. J., Park, D. J., &amp; Sabeti, P. C. (2019). Benchmarking metagenomics tools for taxonomic classification. <em>Cell</em>, 178(4), 779-794.</p>
<p><a href="https://doi.org/10.1186/gb-2012-13-9-r79">[11]</a> Morgan, X. C., Tickle, T. L., Sokol, H., Gevers, D., Devaney, K. L., Ward, D. V., ... &amp; Huttenhower, C. (2012). Dysfunction of the intestinal microbiome in inflammatory bowel disease and treatment. <em>Genome Biology</em>, 13(9), 1-18.</p>
<p><a href="https://www.zymoresearch.de/collections/zymobiomics-microbial-community-standards/products/zymobiomics-microbial-community-standard">[12]</a> Zymo Research Corporation, Irvine, CA, USA. Product D6300, Lot ZRC190633</p>
<p><a href="https://www.zymoresearch.de/collections/zymobiomics-microbial-community-standards/products/zymobiomics-microbial-community-standard-ii-log-distribution">[13]</a> Zymo Research Corporation, Irvine, CA, USA. Product D6310, Lot ZRC190842</p>
<p><a href="https://doi.org/10.1093/gigascience/giz043">[14]</a> Nicholls, S. M., Quick, J. C., Tang, S., &amp; Loman, N. J. (2019). Ultra-deep, long-read nanopore sequencing of mock microbial community standards. <em>Gigascience</em>, 8(5), giz043.</p>
<p><a href="https://doi.org/10.1186/s12859-018-2336-6">[15]</a> Clausen, P. T., Aarestrup, F. M., &amp; Lund, O. (2018). Rapid and precise alignment of raw reads against redundant databases with KMA. <em>BMC bioinformatics</em>, 19(1), 1-8.</p>
<p><a href="https://docs.anaconda.com/">[16]</a> Anaconda Software Distribution. (2020). Anaconda Documentation. <em>Anaconda Inc.</em> Retrieved from https://docs.anaconda.com/</p>
<p><a href="https://snakemake.readthedocs.io/en/stable/">[17]</a> Köster, J., &amp; Rahmann, S. (2012). Snakemake—a scalable bioinformatics workflow engine. Bioinformatics, 28(19), 2520-2522.</p>
<p><a href="https://doi.org/10.1016/S0022-2836(05)80360-2">[18]</a> Altschul, S. F., Gish, W., Miller, W., Myers, E. W., &amp; Lipman, D. J. (1990). Basic local alignment search tool. <em>Journal of molecular biology</em>, 215(3), 403-410.</p>
<p><a href="https://cran.r-project.org/web/packages/PRROC/index.html">[19]</a> Jens Keilwagen, Ivo Grosse and Jan Grau (2014). Area under Precision-Recall Curves for Weighted and Unweighted Data. <em>PLOS ONE</em> (9) 3.</p>
<p><a href="https://www.sciencedirect.com/science/article/pii/S0580951715000148">[20]</a> Dingle, T. C., &amp; MacCannell, D. R. (2015). Molecular strain typing and characterisation of toxigenic Clostridium difficile. <em>Methods in Microbiology</em>, 42, 329-357.</p>
<p><a href="https://pubmlst.org/multilocus-sequence-typing">[21]</a> Public databases for molecular typing and microbial genome diversity. Multi-Locus Sequence Typing. Retrieved from https://pubmlst.org/multilocus-sequence-typing. Last visited on 21/04/2021.</p>
<p><a href="https://doi.org/10.1007/s00203-017-1346-5">[22]</a> Poluektova, E. U., Yunes, R. A., Epiphanova, M. V., Orlova, V. S., &amp; Danilenko, V. N. (2017). The Lactobacillus rhamnosus and Lactobacillus fermentum strains from human biotopes characterized with MLST and toxin-antitoxin gene polymorphism. <em>Archives of microbiology</em>, 199(5), 683-690.</p>
<p><a href="https://doi.org/10.1080/13693780902953886">[23]</a> Meyer, W., Aanensen, D. M., Boekhout, T., Cogliati, M., Diaz, M. R., Esposto, M. C., ... &amp; Kwon-Chung, J. (2009). Consensus multi-locus sequence typing scheme for Cryptococcus neoformans and Cryptococcus gattii. <em>Medical mycology</em>, 47(6), 561-570.</p>
<p><a href="https://doi.org/10.1007/s10068-018-0335-z">[24]</a> Eeom, Y. J., Son, S. Y., Jung, D. H., Hur, M. S., Kim, C. M., Park, S. Y., ... &amp; Park, C. S. (2018). Diversity analysis of Saccharomyces cerevisiae isolated from natural sources by multilocus sequence typing (MLST). <em>Food science and biotechnology</em>, 27(4), 1119-1127.</p>
<p><a href="https://doi.org/10.1186/s12859-021-04089-5">[25]</a> Fan, J., Huang, S., &amp; Chorlton, S. D. (2021). BugSeq: a highly accurate cloud platform for long-read metagenomic analyses. <em>BMC bioinformatics</em>, 22(1), 1-12.</p>
<p><a href="https://doi.org/10.1099/ijsem.0.004107">[26]</a> Zheng, J., Wittouck, S., Salvetti, E., Franz, C. M., Harris, H. M., Mattarelli, P., ... &amp; Lebeer, S. (2020). A taxonomic note on the genus Lactobacillus: Description of 23 novel genera, emended description of the genus Lactobacillus Beijerinck 1901, and union of Lactobacillaceae and Leuconostocaceae. <em>International journal of systematic and evolutionary microbiology</em>, 70(4), 2782-2858.</p>
<p><a href="https://doi.org/10.1007/s10482-019-01354-9">[27]</a> Dunlap, C. A., Bowman, M. J., &amp; Zeigler, D. R. (2020). Promotion of Bacillus subtilis subsp. inaquosorum, Bacillus subtilis subsp. spizizenii and Bacillus subtilis subsp. stercoris to species status. <em>Antonie van Leeuwenhoek</em>, 113(1), 1-12.</p>
<p><a href="https://academic.oup.com/bioinformatics/article/27/1/127/202209">[28]</a> Rosen, G. L., Reichenberger, E. R., &amp; Rosenfeld, A. M. (2011). NBC: the Naive Bayes Classification tool webserver for taxonomic classification of metagenomic reads. <em>Bioinformatics</em>, 27(1), 127-129.</p>
<p><a href="https://academic.oup.com/nar/article/45/4/1649/2674183">[29]</a> Ainsworth, D., Sternberg, M. J., Raczy, C., &amp; Butcher, S. A. (2017). k-SLAM: accurate and ultra-fast taxonomic classification and gene identification for large metagenomic data sets. <em>Nucleic acids research</em>, 45(4), 1649-1656.</p>
<p><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03628-w">[30]</a> Guerrini, V., Louza, F. A., &amp; Rosone, G. (2020). Metagenomic analysis through the extended Burrows-Wheeler transform. <em>BMC bioinformatics</em>, 21(8), 1-25.</p>
<p><a href="https://genome.cshlp.org/content/28/5/751">[31]</a> Corvelo, A., CLARKe, W. E., Robine, N., &amp; Zody, M. C. (2018). taxMaps: comprehensive and highly accurate taxonomic classification of short-read data in reasonable time. <em>Genome research</em>, 28(5), 751-758.</p>
<p><a href="https://doi.org/10.1093/bioinformatics/btx432">[32]</a> Liu, X., Yu, Y., Liu, J., Elliott, C. F., Qian, C., &amp; Liu, J. (2018). A novel data structure to support ultra-fast taxonomic classification of metagenomic sequences with k-mer signatures. <em>Bioinformatics</em>, 34(1), 171-178.</p>
<p><a href="https://doi.org/10.1093/nargab/lqaa009">[33]</a> Liang, Q., Bible, P. W., Liu, Y., Zou, B., &amp; Wei, L. (2020). DeepMicrobes: taxonomic classification for metagenomics with deep learning. <em>NAR Genomics and Bioinformatics</em>, 2(1), lqaa009.</p>
<p><a href="https://doi.org/10.1093/database/baaa062">[34]</a> Schoch, C. L., Ciufo, S., Domrachev, M., Hotton, C. L., Kannan, S., Khovanskaya, R., ... &amp; Karsch-Mizrachi, I. (2020). NCBI Taxonomy: a comprehensive update on curation, resources and tools. <em>Database</em>, 2020.</p>

</body>
</html>
